{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "C0iq3ZXARm5y"
   },
   "outputs": [],
   "source": [
    "#mount drive\n",
    "from google.colab import drive\n",
    "drive.mount('/content/drive')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "zS_4oo9x57DB",
    "outputId": "1293ffbb-b1ac-4104-936c-e4e387669656"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading...\n",
      "From (original): https://drive.google.com/uc?id=1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO\n",
      "From (redirected): https://drive.google.com/uc?id=1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO&confirm=t&uuid=4590e154-1107-4518-86c5-724fcaed1335\n",
      "To: /content/CRISPR.R1.fastq\n",
      "100% 2.67G/2.67G [00:27<00:00, 98.2MB/s]\n",
      "Downloading...\n",
      "From (original): https://drive.google.com/uc?id=1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS\n",
      "From (redirected): https://drive.google.com/uc?id=1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS&confirm=t&uuid=96e228fd-dcd9-4bd8-bd8f-7c5d2a613244\n",
      "To: /content/CRISPR.R2.fastq\n",
      "100% 2.67G/2.67G [00:32<00:00, 81.0MB/s]\n",
      "Downloading...\n",
      "From (original): https://drive.google.com/uc?id=1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar\n",
      "From (redirected): https://drive.google.com/uc?id=1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar&confirm=t&uuid=2c5c7e71-ed56-4892-b39e-448d896b7252\n",
      "To: /content/MM10.tar.gz\n",
      "100% 3.91G/3.91G [00:38<00:00, 101MB/s]\n"
     ]
    }
   ],
   "source": [
    "###### CRISPR sample\n",
    "\n",
    "!gdown 1-96T1PZKA_FQeD_ZK5z9USaHLP3jVdRO\n",
    "!gdown 1-BXGr3XVGtd9Tx6PCHSp4hepK41GGCTS\n",
    "\n",
    "### To make thinngs faster for the student, here is the indexed mouse genome, mm10.\n",
    "### You have to donwload it and decompressed it, you will get both the fasta and the BWT'ed files. Note all have same prefix Mouse<.files>.\n",
    "### Next semester we will cover how specialized computers are often used to solve this aligmnet problems faster.\n",
    "\n",
    "!gdown 1a8CP4P5zkzIBiw1EleqJiSwDW0VZcAar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "32EVYuAGi6uh",
    "outputId": "f5f0598a-3eb5-4fb5-f0ff-fccd74baf5b0"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "bwa is already the newest version (0.7.17-6).\n",
      "0 upgraded, 0 newly installed, 0 to remove and 49 not upgraded.\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "samtools is already the newest version (1.13-4).\n",
      "0 upgraded, 0 newly installed, 0 to remove and 49 not upgraded.\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "bcftools is already the newest version (1.13-1).\n",
      "0 upgraded, 0 newly installed, 0 to remove and 49 not upgraded.\n",
      "Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
      "Get:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
      "Hit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
      "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
      "Get:5 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
      "Get:6 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,190 kB]\n",
      "Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
      "Get:8 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,625 kB]\n",
      "Get:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease [18.1 kB]\n",
      "Hit:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
      "Get:11 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
      "Get:12 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,225 kB]\n",
      "Hit:13 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
      "Get:14 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,523 kB]\n",
      "Get:15 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 Packages [35.0 kB]\n",
      "Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [3,419 kB]\n",
      "Get:17 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,454 kB]\n",
      "Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,514 kB]\n",
      "Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,738 kB]\n",
      "Get:20 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [53.3 kB]\n",
      "Fetched 24.2 MB in 3s (8,004 kB/s)\n",
      "Reading package lists... Done\n",
      "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "The following NEW packages will be installed:\n",
      "  seqtk\n",
      "0 upgraded, 1 newly installed, 0 to remove and 52 not upgraded.\n",
      "Need to get 30.2 kB of archives.\n",
      "After this operation, 85.0 kB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 seqtk amd64 1.3-2 [30.2 kB]\n",
      "Fetched 30.2 kB in 1s (48.8 kB/s)\n",
      "Selecting previously unselected package seqtk.\n",
      "(Reading database ... 123803 files and directories currently installed.)\n",
      "Preparing to unpack .../archives/seqtk_1.3-2_amd64.deb ...\n",
      "Unpacking seqtk (1.3-2) ...\n",
      "Setting up seqtk (1.3-2) ...\n",
      "Processing triggers for man-db (2.10.2-1) ...\n"
     ]
    }
   ],
   "source": [
    "!apt install bwa\n",
    "!apt install samtools\n",
    "!apt install bcftools\n",
    "!apt-get update\n",
    "!apt-get install -y seqtk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "oL65HdS5jDsS",
    "outputId": "0892e3b3-b45d-469f-9c3a-6769e332d254"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM10/\n",
      "MM10/Mouse.sa\n",
      "MM10/Mouse.pac\n",
      "MM10/Mouse.ann\n",
      "MM10/Mouse.fasta\n",
      "MM10/Mouse.bwt\n",
      "MM10/Mouse.amb\n"
     ]
    }
   ],
   "source": [
    "!tar -xvzf MM10.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "zT14HfQkjJAr",
    "outputId": "8c171dc1-a700-4aa5-eb5a-971d08395cc7"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mouse.amb  Mouse.ann  Mouse.bwt  Mouse.fasta  Mouse.pac  Mouse.sa\n"
     ]
    }
   ],
   "source": [
    "!ls MM10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "zGZHBcRRjS1u"
   },
   "outputs": [],
   "source": [
    "#isolating only chromosome 2 for analysis\n",
    "!samtools faidx MM10/Mouse.fasta chr2 > chr2.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "0TolBAaKjYhF",
    "outputId": "3b5dc7bf-0a8e-4296-e5be-1ae853d70972"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 2.25 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=364226448, availableWord=37628116\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 62069712 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 114669760 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 161416816 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 202961696 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 239882848 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 272694448 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 301853424 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 327765984 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 350793104 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 97 iterations.\n",
      "[bwa_index] 187.66 seconds elapse.\n",
      "[bwa_index] Update BWT... 1.20 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.98 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 95.35 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index chr2.fasta\n",
      "[main] Real time: 313.067 sec; CPU: 287.433 sec\n"
     ]
    }
   ],
   "source": [
    "#Indexing chromosme 2\n",
    "!bwa index chr2.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "kj2gzRsTjWKZ",
    "outputId": "1ed148b5-9ead-406a-8c46-902b2fd8d097"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 184, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 234, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (238.14, 59.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.161 CPU sec, 23.672 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 67, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (212, 267, 364)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 668)\n",
      "[M::mem_pestat] mean and std.dev: (258.77, 97.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 820)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.448 CPU sec, 30.930 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 462)\n",
      "[M::mem_pestat] mean and std.dev: (248.92, 59.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 546)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.954 CPU sec, 24.461 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 162, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 228, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 425)\n",
      "[M::mem_pestat] mean and std.dev: (236.90, 57.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 500)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.972 CPU sec, 24.296 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (213, 285, 353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 633)\n",
      "[M::mem_pestat] mean and std.dev: (286.59, 102.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 773)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.300 CPU sec, 29.031 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 13, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (158, 245, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 593)\n",
      "[M::mem_pestat] mean and std.dev: (226.77, 79.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 738)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.387 CPU sec, 24.812 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 86, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 291)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 473)\n",
      "[M::mem_pestat] mean and std.dev: (236.91, 55.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.721 CPU sec, 28.535 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 26, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 246, 273)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (54, 419)\n",
      "[M::mem_pestat] mean and std.dev: (225.13, 55.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.322 CPU sec, 23.726 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 226, 271)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 421)\n",
      "[M::mem_pestat] mean and std.dev: (228.06, 53.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 496)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.317 CPU sec, 30.244 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 213, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 437)\n",
      "[M::mem_pestat] mean and std.dev: (234.71, 61.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.393 CPU sec, 23.828 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (187, 237, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 565)\n",
      "[M::mem_pestat] mean and std.dev: (244.09, 65.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.235 CPU sec, 28.826 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 578)\n",
      "[M::mem_pestat] mean and std.dev: (254.21, 63.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 701)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.746 CPU sec, 27.336 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 231, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (19, 469)\n",
      "[M::mem_pestat] mean and std.dev: (237.83, 65.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 559)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.384 CPU sec, 24.705 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 242, 283)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (55, 435)\n",
      "[M::mem_pestat] mean and std.dev: (243.14, 53.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 511)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.558 CPU sec, 27.241 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 35, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 203, 256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 392)\n",
      "[M::mem_pestat] mean and std.dev: (215.48, 43.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 460)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.313 CPU sec, 22.640 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 20872, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 506)\n",
      "[M::mem_pestat] mean and std.dev: (263.84, 57.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 19.839 CPU sec, 20.479 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47226, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (13, 503)\n",
      "[M::mem_pestat] mean and std.dev: (263.62, 57.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.784 CPU sec, 13.193 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47220, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 507)\n",
      "[M::mem_pestat] mean and std.dev: (264.46, 57.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.239 CPU sec, 11.474 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47418, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 507)\n",
      "[M::mem_pestat] mean and std.dev: (264.51, 57.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.338 CPU sec, 9.194 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47964, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 506)\n",
      "[M::mem_pestat] mean and std.dev: (264.30, 57.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 13.266 CPU sec, 14.189 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 47962, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.66, 56.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.268 CPU sec, 12.560 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47721, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.49, 57.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.329 CPU sec, 12.700 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 45400, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 256, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (12, 507)\n",
      "[M::mem_pestat] mean and std.dev: (264.20, 57.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.902 CPU sec, 13.330 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 46647, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 506)\n",
      "[M::mem_pestat] mean and std.dev: (264.10, 57.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 605)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 9.024 CPU sec, 8.871 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47269, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.14, 56.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 15.389 CPU sec, 18.726 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 47026, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.39, 57.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.774 CPU sec, 12.426 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47373, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (13, 503)\n",
      "[M::mem_pestat] mean and std.dev: (263.73, 57.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.898 CPU sec, 10.877 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47826, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 257, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (264.62, 57.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 10.881 CPU sec, 10.874 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44891, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 256, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (9, 509)\n",
      "[M::mem_pestat] mean and std.dev: (264.36, 57.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.060 CPU sec, 12.254 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46677, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (8, 508)\n",
      "[M::mem_pestat] mean and std.dev: (263.58, 57.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 12.982 CPU sec, 13.395 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41795, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 256, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 504)\n",
      "[M::mem_pestat] mean and std.dev: (263.87, 57.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 602)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 16.753 CPU sec, 17.276 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 4123, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (210, 259, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 510)\n",
      "[M::mem_pestat] mean and std.dev: (264.40, 57.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 610)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.340 CPU sec, 22.723 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 228, 262)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (79, 384)\n",
      "[M::mem_pestat] mean and std.dev: (237.65, 54.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (18, 456)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.150 CPU sec, 27.936 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 242, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 609)\n",
      "[M::mem_pestat] mean and std.dev: (233.15, 91.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 747)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.729 CPU sec, 25.193 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 33, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 587)\n",
      "[M::mem_pestat] mean and std.dev: (245.20, 64.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 716)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.824 CPU sec, 29.223 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 243, 326)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 576)\n",
      "[M::mem_pestat] mean and std.dev: (252.57, 64.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 701)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.185 CPU sec, 28.756 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (189, 228, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 447)\n",
      "[M::mem_pestat] mean and std.dev: (233.95, 58.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 533)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.989 CPU sec, 24.317 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 145, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 250, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (248.87, 69.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.339 CPU sec, 29.894 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 35, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 240, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 418)\n",
      "[M::mem_pestat] mean and std.dev: (228.82, 56.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.653 CPU sec, 23.102 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48, 0, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 248, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 540)\n",
      "[M::mem_pestat] mean and std.dev: (249.46, 51.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 651)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.689 CPU sec, 29.511 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 73, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 249, 338)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 612)\n",
      "[M::mem_pestat] mean and std.dev: (242.31, 65.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 749)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.870 CPU sec, 26.306 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 269)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (62, 407)\n",
      "[M::mem_pestat] mean and std.dev: (239.59, 47.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 476)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.062 CPU sec, 24.373 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 21, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 251, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 497)\n",
      "[M::mem_pestat] mean and std.dev: (253.90, 58.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.279 CPU sec, 27.960 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 236, 285)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 479)\n",
      "[M::mem_pestat] mean and std.dev: (238.64, 62.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 576)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.537 CPU sec, 23.865 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 136, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 249, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 519)\n",
      "[M::mem_pestat] mean and std.dev: (253.63, 63.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 625)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.571 CPU sec, 29.140 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 235, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 418)\n",
      "[M::mem_pestat] mean and std.dev: (224.86, 59.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.292 CPU sec, 28.547 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 222, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 491)\n",
      "[M::mem_pestat] mean and std.dev: (229.81, 50.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 588)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.738 CPU sec, 28.117 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 265, 1257)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3379)\n",
      "[M::mem_pestat] mean and std.dev: (695.85, 880.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4440)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 50.186 CPU sec, 51.859 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 79, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 233, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 538)\n",
      "[M::mem_pestat] mean and std.dev: (242.21, 63.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 649)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.185 CPU sec, 30.094 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 237, 1348)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3638)\n",
      "[M::mem_pestat] mean and std.dev: (625.19, 928.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4783)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 50.762 CPU sec, 52.363 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 65, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (220, 254, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 481)\n",
      "[M::mem_pestat] mean and std.dev: (249.14, 59.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.654 CPU sec, 22.813 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 33, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (224, 259, 748)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1796)\n",
      "[M::mem_pestat] mean and std.dev: (417.77, 368.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2320)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 38.612 CPU sec, 39.233 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 238, 284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (26, 456)\n",
      "[M::mem_pestat] mean and std.dev: (234.60, 69.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 542)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.375 CPU sec, 24.763 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 581)\n",
      "[M::mem_pestat] mean and std.dev: (247.59, 71.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 705)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.284 CPU sec, 31.830 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 243, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 244, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 477)\n",
      "[M::mem_pestat] mean and std.dev: (245.08, 56.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.678 CPU sec, 25.029 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 107, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 250, 322)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 560)\n",
      "[M::mem_pestat] mean and std.dev: (251.70, 70.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 679)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.868 CPU sec, 26.263 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 40, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (206, 260, 328)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 572)\n",
      "[M::mem_pestat] mean and std.dev: (253.03, 59.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 694)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.908 CPU sec, 28.486 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 90, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (185, 218, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
      "[M::mem_pestat] mean and std.dev: (226.98, 68.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.661 CPU sec, 22.944 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 25, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 231, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 459)\n",
      "[M::mem_pestat] mean and std.dev: (243.68, 59.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 26.286 CPU sec, 26.953 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 256, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 548)\n",
      "[M::mem_pestat] mean and std.dev: (240.62, 53.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 664)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.606 CPU sec, 30.432 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 238, 738)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1810)\n",
      "[M::mem_pestat] mean and std.dev: (249.00, 112.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2346)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 40.314 CPU sec, 40.922 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 40, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 226, 274)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 422)\n",
      "[M::mem_pestat] mean and std.dev: (241.90, 59.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 496)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.302 CPU sec, 25.873 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 102, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 233, 346)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 638)\n",
      "[M::mem_pestat] mean and std.dev: (239.11, 69.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 784)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.073 CPU sec, 27.323 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 19, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (191, 235, 273)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 437)\n",
      "[M::mem_pestat] mean and std.dev: (227.00, 62.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 519)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.475 CPU sec, 28.090 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 416, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 235, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 485)\n",
      "[M::mem_pestat] mean and std.dev: (245.35, 65.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.754 CPU sec, 24.126 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 30, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 215, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 482)\n",
      "[M::mem_pestat] mean and std.dev: (236.00, 74.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.098 CPU sec, 30.708 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 254, 298)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 496)\n",
      "[M::mem_pestat] mean and std.dev: (241.85, 76.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.451 CPU sec, 27.739 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 57, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 362)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 686)\n",
      "[M::mem_pestat] mean and std.dev: (253.54, 70.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 848)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 31.610 CPU sec, 32.311 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 241, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
      "[M::mem_pestat] mean and std.dev: (258.70, 74.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 626)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.147 CPU sec, 28.815 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 277, 1367)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3691)\n",
      "[M::mem_pestat] mean and std.dev: (655.47, 674.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4853)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 52.309 CPU sec, 55.766 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 17, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (233, 286, 363)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 623)\n",
      "[M::mem_pestat] mean and std.dev: (255.29, 66.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 753)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.151 CPU sec, 25.435 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 26, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (213, 279, 1414)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3816)\n",
      "[M::mem_pestat] mean and std.dev: (702.12, 758.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5017)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 53.353 CPU sec, 54.312 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (183, 261, 1394)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3816)\n",
      "[M::mem_pestat] mean and std.dev: (688.23, 757.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5027)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 52.763 CPU sec, 53.676 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 68, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 237, 288)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (24, 464)\n",
      "[M::mem_pestat] mean and std.dev: (239.67, 58.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 552)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.875 CPU sec, 28.264 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 225, 282)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 466)\n",
      "[M::mem_pestat] mean and std.dev: (234.64, 62.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 558)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.754 CPU sec, 29.349 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 282, 1785)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4941)\n",
      "[M::mem_pestat] mean and std.dev: (786.83, 887.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6519)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 61.688 CPU sec, 62.845 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 37, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 222, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 490)\n",
      "[M::mem_pestat] mean and std.dev: (237.70, 59.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 587)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.829 CPU sec, 24.116 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 83, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (230, 294, 376)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 668)\n",
      "[M::mem_pestat] mean and std.dev: (282.12, 98.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 814)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.853 CPU sec, 29.508 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 127, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 260, 338)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 600)\n",
      "[M::mem_pestat] mean and std.dev: (252.64, 71.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 731)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.087 CPU sec, 27.597 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 28, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (189, 210, 254)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (59, 384)\n",
      "[M::mem_pestat] mean and std.dev: (212.20, 48.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 449)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.782 CPU sec, 24.074 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 148, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 255, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 593)\n",
      "[M::mem_pestat] mean and std.dev: (249.18, 64.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 723)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 34.572 CPU sec, 35.759 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 350, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 234, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 458)\n",
      "[M::mem_pestat] mean and std.dev: (240.85, 58.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.984 CPU sec, 24.320 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 130, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 238, 334)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 616)\n",
      "[M::mem_pestat] mean and std.dev: (227.04, 77.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 757)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.209 CPU sec, 29.770 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 94, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 467)\n",
      "[M::mem_pestat] mean and std.dev: (249.47, 66.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.354 CPU sec, 28.007 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (186, 219, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
      "[M::mem_pestat] mean and std.dev: (221.72, 65.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 598)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.085 CPU sec, 23.371 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 219, 276)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 440)\n",
      "[M::mem_pestat] mean and std.dev: (218.87, 64.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 522)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.506 CPU sec, 29.034 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 271, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (264.93, 65.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 637)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.638 CPU sec, 25.065 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (159, 220, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 561)\n",
      "[M::mem_pestat] mean and std.dev: (209.25, 76.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 695)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.443 CPU sec, 28.553 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 63, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (212, 243, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 563)\n",
      "[M::mem_pestat] mean and std.dev: (243.98, 55.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 680)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.114 CPU sec, 29.299 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 67, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (215, 273, 408)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 794)\n",
      "[M::mem_pestat] mean and std.dev: (275.18, 114.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 987)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.238 CPU sec, 25.449 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 23, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 271, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 692)\n",
      "[M::mem_pestat] mean and std.dev: (263.45, 103.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 858)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.857 CPU sec, 23.017 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 77, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 292)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 476)\n",
      "[M::mem_pestat] mean and std.dev: (239.12, 61.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 20.906 CPU sec, 20.920 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 12, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 248, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 505)\n",
      "[M::mem_pestat] mean and std.dev: (242.33, 48.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 606)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.149 CPU sec, 22.370 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 386, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 250, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (3, 513)\n",
      "[M::mem_pestat] mean and std.dev: (257.69, 58.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 615)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.416 CPU sec, 23.635 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 64, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (197, 228, 300)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (236.80, 69.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 609)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.098 CPU sec, 23.162 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 40, 0, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 557)\n",
      "[M::mem_pestat] mean and std.dev: (225.45, 41.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 676)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.740 CPU sec, 24.975 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 240, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 425)\n",
      "[M::mem_pestat] mean and std.dev: (237.85, 53.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 500)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.992 CPU sec, 21.953 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 251, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 525)\n",
      "[M::mem_pestat] mean and std.dev: (251.62, 65.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 633)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.293 CPU sec, 24.478 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 38, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 339)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 617)\n",
      "[M::mem_pestat] mean and std.dev: (236.68, 75.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 756)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.819 CPU sec, 24.082 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 231, 284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (59, 434)\n",
      "[M::mem_pestat] mean and std.dev: (242.38, 56.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 509)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.553 CPU sec, 21.658 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 230, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (240.94, 64.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 672)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.882 CPU sec, 26.097 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 11, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (232, 268, 351)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 589)\n",
      "[M::mem_pestat] mean and std.dev: (263.09, 75.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 708)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.098 CPU sec, 24.301 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 78, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 233, 282)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 450)\n",
      "[M::mem_pestat] mean and std.dev: (233.42, 56.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.030 CPU sec, 24.295 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 77, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 324)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 572)\n",
      "[M::mem_pestat] mean and std.dev: (251.89, 75.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 696)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.330 CPU sec, 24.626 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 238, 285)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (33, 453)\n",
      "[M::mem_pestat] mean and std.dev: (235.13, 55.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 537)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.514 CPU sec, 21.653 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 239, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (32, 447)\n",
      "[M::mem_pestat] mean and std.dev: (236.52, 63.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 530)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.565 CPU sec, 23.802 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 224, 305)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 515)\n",
      "[M::mem_pestat] mean and std.dev: (234.22, 63.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.828 CPU sec, 23.006 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 268, 345)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 639)\n",
      "[M::mem_pestat] mean and std.dev: (261.27, 86.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 786)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.722 CPU sec, 25.941 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (220, 284, 1441)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3883)\n",
      "[M::mem_pestat] mean and std.dev: (713.05, 909.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5104)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 51.541 CPU sec, 52.091 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 38, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 239, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 499)\n",
      "[M::mem_pestat] mean and std.dev: (226.56, 70.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.713 CPU sec, 25.022 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 40, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 216, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 499)\n",
      "[M::mem_pestat] mean and std.dev: (234.69, 76.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.660 CPU sec, 23.913 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 77, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 230, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 549)\n",
      "[M::mem_pestat] mean and std.dev: (247.77, 80.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 665)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.179 CPU sec, 25.553 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 26, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (226, 254, 274)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (130, 370)\n",
      "[M::mem_pestat] mean and std.dev: (253.48, 37.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (82, 418)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.610 CPU sec, 30.880 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 137, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 240, 288)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 462)\n",
      "[M::mem_pestat] mean and std.dev: (243.04, 59.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.343 CPU sec, 24.580 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 233, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 517)\n",
      "[M::mem_pestat] mean and std.dev: (231.66, 65.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 625)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.548 CPU sec, 23.794 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 69, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 226, 290)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 490)\n",
      "[M::mem_pestat] mean and std.dev: (234.42, 74.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.140 CPU sec, 23.331 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 105, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 257, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (249.04, 64.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 649)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.886 CPU sec, 24.200 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (236, 309, 3383)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9677)\n",
      "[M::mem_pestat] mean and std.dev: (1517.14, 1751.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12824)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 98.640 CPU sec, 99.962 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 21, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (223, 241, 294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (81, 436)\n",
      "[M::mem_pestat] mean and std.dev: (250.10, 45.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (10, 507)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.486 CPU sec, 24.768 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 66, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 229, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (50, 425)\n",
      "[M::mem_pestat] mean and std.dev: (233.60, 52.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 500)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.403 CPU sec, 22.657 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 5, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 19.715 CPU sec, 19.852 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (214, 325, 1627)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4453)\n",
      "[M::mem_pestat] mean and std.dev: (805.23, 931.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5866)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 54.416 CPU sec, 55.092 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 207, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 456)\n",
      "[M::mem_pestat] mean and std.dev: (218.67, 57.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.040 CPU sec, 24.376 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 178, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 242, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 542)\n",
      "[M::mem_pestat] mean and std.dev: (249.28, 71.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 656)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.899 CPU sec, 25.184 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 271, 323)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 553)\n",
      "[M::mem_pestat] mean and std.dev: (258.34, 83.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 668)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.623 CPU sec, 23.772 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 76, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 215, 268)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 424)\n",
      "[M::mem_pestat] mean and std.dev: (222.74, 54.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 502)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.188 CPU sec, 23.465 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 20, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 243, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 482)\n",
      "[M::mem_pestat] mean and std.dev: (245.00, 58.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 575)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.438 CPU sec, 23.656 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 294, 449)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 947)\n",
      "[M::mem_pestat] mean and std.dev: (254.27, 83.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1196)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.744 CPU sec, 28.027 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 104, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 239, 299)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 507)\n",
      "[M::mem_pestat] mean and std.dev: (242.38, 72.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.966 CPU sec, 24.251 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 20, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 234, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 559)\n",
      "[M::mem_pestat] mean and std.dev: (242.15, 72.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 680)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.233 CPU sec, 24.437 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 68, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (185, 220, 281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
      "[M::mem_pestat] mean and std.dev: (213.68, 66.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.595 CPU sec, 24.829 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 21, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 210, 263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (74, 389)\n",
      "[M::mem_pestat] mean and std.dev: (222.60, 52.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (11, 452)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.848 CPU sec, 24.035 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 58, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 230, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 594)\n",
      "[M::mem_pestat] mean and std.dev: (224.96, 62.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 726)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.052 CPU sec, 33.056 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 27, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (216, 251, 272)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (104, 384)\n",
      "[M::mem_pestat] mean and std.dev: (239.23, 49.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (40, 440)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.596 CPU sec, 23.683 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 171, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 251, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 515)\n",
      "[M::mem_pestat] mean and std.dev: (246.57, 58.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 619)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.305 CPU sec, 24.618 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 51, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 233, 280)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (40, 440)\n",
      "[M::mem_pestat] mean and std.dev: (243.45, 68.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 520)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.390 CPU sec, 23.640 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (218, 265, 301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (52, 467)\n",
      "[M::mem_pestat] mean and std.dev: (269.11, 68.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 550)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.696 CPU sec, 22.881 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 248, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 549)\n",
      "[M::mem_pestat] mean and std.dev: (249.86, 58.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 664)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.024 CPU sec, 23.211 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 38, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (191, 206, 273)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (27, 437)\n",
      "[M::mem_pestat] mean and std.dev: (221.88, 53.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 519)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.206 CPU sec, 22.200 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 65, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 247, 300)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 500)\n",
      "[M::mem_pestat] mean and std.dev: (235.04, 56.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 600)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.953 CPU sec, 22.975 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 230, 268)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (64, 404)\n",
      "[M::mem_pestat] mean and std.dev: (236.94, 55.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 472)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.966 CPU sec, 23.106 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 66, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 254, 604)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1410)\n",
      "[M::mem_pestat] mean and std.dev: (305.57, 206.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1813)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.746 CPU sec, 31.138 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 255, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (23, 488)\n",
      "[M::mem_pestat] mean and std.dev: (245.24, 63.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.378 CPU sec, 23.552 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 227, 284)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 462)\n",
      "[M::mem_pestat] mean and std.dev: (225.48, 60.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 551)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.445 CPU sec, 21.515 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 69, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 246, 316)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 548)\n",
      "[M::mem_pestat] mean and std.dev: (241.30, 58.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 664)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.107 CPU sec, 23.140 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 112, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 244, 291)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 473)\n",
      "[M::mem_pestat] mean and std.dev: (239.08, 62.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.325 CPU sec, 21.397 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 226, 277)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (70, 415)\n",
      "[M::mem_pestat] mean and std.dev: (236.00, 56.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 484)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.371 CPU sec, 22.490 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 17, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (211, 284, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 529)\n",
      "[M::mem_pestat] mean and std.dev: (256.40, 56.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 635)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.340 CPU sec, 24.632 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 10, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (182, 221, 243)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (60, 365)\n",
      "[M::mem_pestat] mean and std.dev: (205.78, 35.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 426)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.867 CPU sec, 21.976 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 358, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 247, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 485)\n",
      "[M::mem_pestat] mean and std.dev: (248.39, 64.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.410 CPU sec, 23.594 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 78, 0, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 229, 274)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 426)\n",
      "[M::mem_pestat] mean and std.dev: (224.29, 57.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 502)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 20.000 CPU sec, 19.969 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 201, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (18, 433)\n",
      "[M::mem_pestat] mean and std.dev: (214.17, 60.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.910 CPU sec, 22.043 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (183, 214, 241)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (67, 357)\n",
      "[M::mem_pestat] mean and std.dev: (216.50, 55.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 439)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.508 CPU sec, 22.668 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (256.74, 59.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 652)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.369 CPU sec, 24.680 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 760, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 250, 304)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 502)\n",
      "[M::mem_pestat] mean and std.dev: (259.69, 61.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 601)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.978 CPU sec, 23.185 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 125, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 268, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 597)\n",
      "[M::mem_pestat] mean and std.dev: (253.15, 74.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 729)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.968 CPU sec, 24.169 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 227, 276)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (75, 410)\n",
      "[M::mem_pestat] mean and std.dev: (235.16, 55.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (8, 477)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.020 CPU sec, 21.041 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 158, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 254, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (242.08, 66.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 637)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.303 CPU sec, 22.395 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 174, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 241, 306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 514)\n",
      "[M::mem_pestat] mean and std.dev: (255.11, 71.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 618)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 27.170 CPU sec, 30.242 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 250, 322)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 564)\n",
      "[M::mem_pestat] mean and std.dev: (251.40, 62.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 685)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.226 CPU sec, 24.448 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 530)\n",
      "[M::mem_pestat] mean and std.dev: (250.23, 62.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 640)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.987 CPU sec, 24.101 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 64, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 292)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 476)\n",
      "[M::mem_pestat] mean and std.dev: (247.98, 58.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.753 CPU sec, 23.883 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 244, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (241.42, 56.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.692 CPU sec, 22.899 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (214, 247, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 514)\n",
      "[M::mem_pestat] mean and std.dev: (246.97, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 614)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.394 CPU sec, 21.454 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 437)\n",
      "[M::mem_pestat] mean and std.dev: (241.49, 63.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.999 CPU sec, 22.028 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 50, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 236, 311)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (250.02, 70.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 635)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.915 CPU sec, 24.072 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 232, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (66, 421)\n",
      "[M::mem_pestat] mean and std.dev: (243.35, 61.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.892 CPU sec, 23.144 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 222, 282)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (30, 450)\n",
      "[M::mem_pestat] mean and std.dev: (229.90, 65.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.207 CPU sec, 23.442 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 243, 278)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (44, 434)\n",
      "[M::mem_pestat] mean and std.dev: (241.82, 53.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 512)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.857 CPU sec, 23.014 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 18, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 212, 252)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (96, 356)\n",
      "[M::mem_pestat] mean and std.dev: (217.39, 54.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 435)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.833 CPU sec, 22.117 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 22, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (141, 200, 239)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 435)\n",
      "[M::mem_pestat] mean and std.dev: (199.00, 57.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 533)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.415 CPU sec, 21.431 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 33, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (225, 266, 366)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 648)\n",
      "[M::mem_pestat] mean and std.dev: (274.29, 94.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 789)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.741 CPU sec, 22.869 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 89, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (28, 458)\n",
      "[M::mem_pestat] mean and std.dev: (240.88, 61.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.574 CPU sec, 23.861 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 271, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 670)\n",
      "[M::mem_pestat] mean and std.dev: (260.72, 87.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 825)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.763 CPU sec, 25.054 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 103, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 237, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (6, 491)\n",
      "[M::mem_pestat] mean and std.dev: (238.83, 62.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 588)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.873 CPU sec, 23.037 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 59, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 235, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (233.81, 70.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 608)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.097 CPU sec, 24.379 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (179, 235, 386)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 800)\n",
      "[M::mem_pestat] mean and std.dev: (215.79, 108.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1007)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.242 CPU sec, 24.266 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 275, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 584)\n",
      "[M::mem_pestat] mean and std.dev: (253.53, 64.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 711)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.167 CPU sec, 24.261 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 15, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 226, 256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (94, 364)\n",
      "[M::mem_pestat] mean and std.dev: (233.80, 54.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (15, 452)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.691 CPU sec, 22.977 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (223, 264, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 493)\n",
      "[M::mem_pestat] mean and std.dev: (249.05, 46.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.767 CPU sec, 24.008 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 100, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 256, 304)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (4, 504)\n",
      "[M::mem_pestat] mean and std.dev: (249.59, 63.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.268 CPU sec, 23.391 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 237, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 475)\n",
      "[M::mem_pestat] mean and std.dev: (233.37, 56.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.253 CPU sec, 22.488 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 250, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 255, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (253.96, 65.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.920 CPU sec, 21.933 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 62, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (8, 488)\n",
      "[M::mem_pestat] mean and std.dev: (240.73, 70.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 584)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.000 CPU sec, 24.178 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 242, 306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 512)\n",
      "[M::mem_pestat] mean and std.dev: (244.71, 67.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 615)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.411 CPU sec, 34.110 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 232, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (11, 486)\n",
      "[M::mem_pestat] mean and std.dev: (238.19, 57.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 581)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.710 CPU sec, 25.051 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 26, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 249, 292)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 476)\n",
      "[M::mem_pestat] mean and std.dev: (233.00, 71.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.277 CPU sec, 23.450 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 228, 290)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (17, 472)\n",
      "[M::mem_pestat] mean and std.dev: (239.80, 63.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 563)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.071 CPU sec, 24.312 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 132, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 249, 312)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (250.65, 64.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.471 CPU sec, 22.628 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 140, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 238, 325)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 575)\n",
      "[M::mem_pestat] mean and std.dev: (230.57, 57.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 700)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.805 CPU sec, 23.978 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44, 0, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 226, 279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (42, 437)\n",
      "[M::mem_pestat] mean and std.dev: (239.45, 61.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 516)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.586 CPU sec, 22.751 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 70, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 524)\n",
      "[M::mem_pestat] mean and std.dev: (242.09, 61.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 632)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.642 CPU sec, 23.771 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 227, 276)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 428)\n",
      "[M::mem_pestat] mean and std.dev: (238.18, 50.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 504)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.945 CPU sec, 24.209 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 239, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (77, 392)\n",
      "[M::mem_pestat] mean and std.dev: (230.71, 47.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (14, 455)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 21.587 CPU sec, 21.726 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (230, 257, 317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 491)\n",
      "[M::mem_pestat] mean and std.dev: (256.43, 57.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.998 CPU sec, 23.065 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 252, 297)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (3, 493)\n",
      "[M::mem_pestat] mean and std.dev: (237.72, 60.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 591)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.002 CPU sec, 22.029 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 114, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 248, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 525)\n",
      "[M::mem_pestat] mean and std.dev: (239.28, 69.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 633)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.575 CPU sec, 22.600 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 252, 294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (15, 480)\n",
      "[M::mem_pestat] mean and std.dev: (249.54, 51.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 573)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.853 CPU sec, 22.974 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 56, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 246, 296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 490)\n",
      "[M::mem_pestat] mean and std.dev: (245.64, 62.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 587)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.435 CPU sec, 23.582 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 106, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 239, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 479)\n",
      "[M::mem_pestat] mean and std.dev: (236.56, 58.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.032 CPU sec, 24.146 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 19, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 258, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 527)\n",
      "[M::mem_pestat] mean and std.dev: (246.58, 77.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 636)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.607 CPU sec, 22.698 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 303, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 241, 298)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 502)\n",
      "[M::mem_pestat] mean and std.dev: (244.19, 68.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 604)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.453 CPU sec, 23.579 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 112, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 254, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 584)\n",
      "[M::mem_pestat] mean and std.dev: (254.82, 69.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 711)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.028 CPU sec, 25.245 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 84, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 253, 353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 641)\n",
      "[M::mem_pestat] mean and std.dev: (263.56, 94.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 785)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.549 CPU sec, 23.614 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 92, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 223, 278)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (38, 438)\n",
      "[M::mem_pestat] mean and std.dev: (229.08, 54.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 518)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 30.413 CPU sec, 31.760 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 164, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 255, 342)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 626)\n",
      "[M::mem_pestat] mean and std.dev: (251.41, 82.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 768)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.020 CPU sec, 25.134 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 115, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 236, 303)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 509)\n",
      "[M::mem_pestat] mean and std.dev: (236.29, 59.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 612)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.218 CPU sec, 24.294 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 173, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 231, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (22, 467)\n",
      "[M::mem_pestat] mean and std.dev: (240.95, 59.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 25.932 CPU sec, 26.035 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 100, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (10, 485)\n",
      "[M::mem_pestat] mean and std.dev: (240.90, 61.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 580)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.522 CPU sec, 23.691 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 132, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 238, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 508)\n",
      "[M::mem_pestat] mean and std.dev: (244.71, 67.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 611)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 29.558 CPU sec, 34.484 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 160, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 247, 299)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5, 495)\n",
      "[M::mem_pestat] mean and std.dev: (249.74, 62.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 593)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.433 CPU sec, 24.600 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 85, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 224, 264)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (69, 394)\n",
      "[M::mem_pestat] mean and std.dev: (236.35, 56.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (4, 461)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.223 CPU sec, 23.386 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 93, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (188, 203, 254)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (56, 386)\n",
      "[M::mem_pestat] mean and std.dev: (220.40, 49.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 452)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.913 CPU sec, 23.091 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 72, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (201, 230, 280)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (43, 438)\n",
      "[M::mem_pestat] mean and std.dev: (241.07, 56.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 517)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.046 CPU sec, 23.058 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 240, 275)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (116, 381)\n",
      "[M::mem_pestat] mean and std.dev: (250.47, 46.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (62, 438)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.771 CPU sec, 22.965 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1701, 10, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 243, 285)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (36, 451)\n",
      "[M::mem_pestat] mean and std.dev: (249.99, 49.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 534)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (610, 700, 717)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (396, 931)\n",
      "[M::mem_pestat] mean and std.dev: (667.60, 59.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (289, 1038)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 20.843 CPU sec, 20.876 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (255, 26456, 728, 134)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5935, 7042, 9226)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15808)\n",
      "[M::mem_pestat] mean and std.dev: (7037.24, 1985.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19099)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 244, 286)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (34, 454)\n",
      "[M::mem_pestat] mean and std.dev: (248.86, 47.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 538)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (853, 4550, 8569)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 24001)\n",
      "[M::mem_pestat] mean and std.dev: (4486.34, 3465.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 31717)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5829, 8417, 8805)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14757)\n",
      "[M::mem_pestat] mean and std.dev: (6357.01, 3083.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18693)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 8.911 CPU sec, 8.935 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 8324, 559, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (5685, 5708, 5740)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (5575, 5850)\n",
      "[M::mem_pestat] mean and std.dev: (5694.64, 45.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (5513, 5905)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 244, 289)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (31, 461)\n",
      "[M::mem_pestat] mean and std.dev: (246.99, 47.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 547)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (689, 767, 3145)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8057)\n",
      "[M::mem_pestat] mean and std.dev: (1593.57, 1394.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10513)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (9310, 9325, 9378)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (9174, 9514)\n",
      "[M::mem_pestat] mean and std.dev: (9336.54, 39.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (9106, 9582)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 52.304 CPU sec, 52.671 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.593 CPU sec, 11.540 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.898 CPU sec, 11.898 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 5, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.800 CPU sec, 11.822 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 5, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 11.003 CPU sec, 10.901 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 298, 10, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 255, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 583)\n",
      "[M::mem_pestat] mean and std.dev: (261.42, 72.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 710)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (276, 717, 1063)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2637)\n",
      "[M::mem_pestat] mean and std.dev: (669.67, 625.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3424)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 20.432 CPU sec, 20.481 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 6295, 203, 22)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (4679, 5916, 8665)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16637)\n",
      "[M::mem_pestat] mean and std.dev: (5639.33, 2488.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 20623)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 300, 871)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2169)\n",
      "[M::mem_pestat] mean and std.dev: (398.91, 382.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2818)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1592, 4042, 6612)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16652)\n",
      "[M::mem_pestat] mean and std.dev: (4318.67, 3002.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21672)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1171, 2855, 7353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 19717)\n",
      "[M::mem_pestat] mean and std.dev: (4050.59, 3183.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 25899)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 32.228 CPU sec, 32.381 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (77, 41597, 1215, 96)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (2483, 5461, 7319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16991)\n",
      "[M::mem_pestat] mean and std.dev: (5062.60, 2795.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21827)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (227, 311, 1017)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2597)\n",
      "[M::mem_pestat] mean and std.dev: (499.95, 528.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3387)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1560, 3597, 6164)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15372)\n",
      "[M::mem_pestat] mean and std.dev: (3993.12, 2800.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 19976)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1878, 4388, 7563)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 18933)\n",
      "[M::mem_pestat] mean and std.dev: (4636.20, 3018.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 24618)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 13.344 CPU sec, 13.419 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (26, 12969, 449, 27)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1055, 3323, 7046)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 19028)\n",
      "[M::mem_pestat] mean and std.dev: (4029.31, 3163.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 25019)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (227, 309, 968)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2450)\n",
      "[M::mem_pestat] mean and std.dev: (476.39, 487.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3191)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1429, 3340, 5771)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14455)\n",
      "[M::mem_pestat] mean and std.dev: (3873.27, 2791.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18797)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1677, 4695, 7916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 20394)\n",
      "[M::mem_pestat] mean and std.dev: (4646.81, 3049.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 26633)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 100000 reads in 28.855 CPU sec, 28.938 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 350, 15, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 249, 318)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (250.10, 68.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 672)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (164, 340, 1511)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4205)\n",
      "[M::mem_pestat] mean and std.dev: (767.80, 869.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5552)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.854 CPU sec, 22.838 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 429, 29, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 251, 326)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 578)\n",
      "[M::mem_pestat] mean and std.dev: (242.62, 77.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 704)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (401, 1006, 1384)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3350)\n",
      "[M::mem_pestat] mean and std.dev: (912.25, 589.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4333)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 43.117 CPU sec, 43.368 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 309, 11, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 356)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 668)\n",
      "[M::mem_pestat] mean and std.dev: (253.90, 90.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 824)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (278, 1573, 6371)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 18557)\n",
      "[M::mem_pestat] mean and std.dev: (2603.55, 2806.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 24650)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.497 CPU sec, 24.617 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 393, 11, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 251, 310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 530)\n",
      "[M::mem_pestat] mean and std.dev: (245.13, 61.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 640)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (187, 337, 505)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1141)\n",
      "[M::mem_pestat] mean and std.dev: (285.22, 126.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1459)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.638 CPU sec, 22.662 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 352, 19, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 337)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 611)\n",
      "[M::mem_pestat] mean and std.dev: (248.12, 74.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 748)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (279, 806, 1684)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4494)\n",
      "[M::mem_pestat] mean and std.dev: (822.61, 613.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5899)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 53.956 CPU sec, 54.330 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 215, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 254, 336)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 608)\n",
      "[M::mem_pestat] mean and std.dev: (253.20, 73.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 744)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 24.799 CPU sec, 25.139 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 289, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 248, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 534)\n",
      "[M::mem_pestat] mean and std.dev: (253.85, 69.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 644)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.062 CPU sec, 23.642 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 381, 27, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 245, 305)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 515)\n",
      "[M::mem_pestat] mean and std.dev: (243.43, 65.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 620)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (393, 846, 1527)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3795)\n",
      "[M::mem_pestat] mean and std.dev: (968.20, 843.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4929)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 51.195 CPU sec, 57.170 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 428, 26, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 248, 314)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 542)\n",
      "[M::mem_pestat] mean and std.dev: (255.63, 70.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 656)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 373, 905)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2323)\n",
      "[M::mem_pestat] mean and std.dev: (548.32, 509.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3032)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 35.813 CPU sec, 36.350 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 257, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 231, 293)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (14, 479)\n",
      "[M::mem_pestat] mean and std.dev: (243.10, 65.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 572)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 22.618 CPU sec, 22.945 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 398, 22, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 241, 308)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 524)\n",
      "[M::mem_pestat] mean and std.dev: (248.99, 67.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 632)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 645, 1356)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3672)\n",
      "[M::mem_pestat] mean and std.dev: (559.95, 479.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4830)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 43.780 CPU sec, 44.926 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 251, 9, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 253, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (253.26, 67.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 652)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 23.249 CPU sec, 23.915 real sec\n",
      "[M::process] read 12220 sequences (1222000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 597, 35, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (199, 241, 306)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
      "[M::mem_pestat] mean and std.dev: (242.73, 66.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 627)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (239, 311, 882)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2168)\n",
      "[M::mem_pestat] mean and std.dev: (542.88, 453.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2811)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 37.381 CPU sec, 38.516 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 0, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 12220 reads in 2.154 CPU sec, 2.263 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem chr2.fasta CRISPR.R1.fastq CRISPR.R2.fastq\n",
      "[main] Real time: 6210.045 sec; CPU: 6093.023 sec\n"
     ]
    }
   ],
   "source": [
    "# aligning sequencing reads to a reference genome.\n",
    "!bwa mem chr2.fasta CRISPR.R1.fastq CRISPR.R2.fastq > aligned_reads_chr2.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "rj4ZYZyQjmfi",
    "outputId": "e6a1533e-ca81-464b-c739-7cba42d07b34"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 7 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "#converting to a BAM file, and sort\n",
    "!samtools view -Sb aligned_reads_chr2.sam > aligned_reads_chr2.bam\n",
    "!samtools sort aligned_reads_chr2.bam -o sorted_reads_chr2.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "SM6UZxkYjqyZ"
   },
   "outputs": [],
   "source": [
    "#indexing the sorted file\n",
    "!samtools index sorted_reads_chr2.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "XPggTde2kAkc",
    "outputId": "10e55716-8ce2-425f-bb8a-5fc55eba0ab8"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n"
     ]
    }
   ],
   "source": [
    "#Identified genetic variations compared to the reference genome\n",
    "!bcftools mpileup -Ou -f chr2.fasta sorted_reads_chr2.bam | bcftools call -mv -Oz -o variants_chr2.vcf.gz #identify potential variants (SNPs, indels) in your aligned reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "K5rzLe8mkEn0",
    "outputId": "ca9dd08b-6ba1-418b-88fa-d59335d9cdc8"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::hts_open_format] Failed to open file \"MM10/Mouse.gtf\" : No such file or directory\n",
      "Failed to open MM10/Mouse.gtf\n"
     ]
    }
   ],
   "source": [
    "!bcftools annotate -a MM10/Mouse.gtf -c CHROM,POS,REF,ALT,QUAL,DP -Oz -o annotated_variants_chr2.vcf.gz filtered_variants_chr2.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "srJhEk9akOHu"
   },
   "outputs": [],
   "source": [
    "#improving the accuracy by removing low-quality reads\n",
    "!bcftools filter -i 'QUAL>20 && DP>10' variants_chr2.vcf.gz -Oz -o filtered_variants_chr2.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "zYAFQdm_kP3R",
    "outputId": "4bae1d52-b1b0-46e7-b395-f62552c16b8f"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting cyvcf2\n",
      "  Downloading cyvcf2-0.31.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.5 kB)\n",
      "Requirement already satisfied: numpy>=1.16.0 in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (1.26.4)\n",
      "Collecting coloredlogs (from cyvcf2)\n",
      "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n",
      "Requirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (8.1.7)\n",
      "Collecting humanfriendly>=9.1 (from coloredlogs->cyvcf2)\n",
      "  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n",
      "Downloading cyvcf2-0.31.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.8 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.8/6.8 MB\u001b[0m \u001b[31m38.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m3.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m6.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: humanfriendly, coloredlogs, cyvcf2\n",
      "Successfully installed coloredlogs-15.0.1 cyvcf2-0.31.1 humanfriendly-10.0\n"
     ]
    }
   ],
   "source": [
    "!pip install cyvcf2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "B7rGwStKkXCH"
   },
   "outputs": [],
   "source": [
    "!bcftools view variants_chr2.vcf.gz -Ov > filtered_variants_chr2.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 424
    },
    "id": "Qrk0KegZkj1n",
    "outputId": "35e2e88a-eeb3-43b9-bce7-b17a5f3cb3dc"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"df\",\n  \"rows\": 70683,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54986788,\n        \"min\": 3051590,\n        \"max\": 182002086,\n        \"num_unique_values\": 70674,\n        \"samples\": [\n          84317243\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 482,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 545,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 55.72852422128099,\n        \"min\": 3.0108699798583984,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 29713,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-d0ed8f3f-641a-4a7d-a1fa-0955dae81e42\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>52.203800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>45.253502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.249500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>55.332600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>59.256699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70678</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70679</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70680</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70681</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70682</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>5.826790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70683 rows × 8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d0ed8f3f-641a-4a7d-a1fa-0955dae81e42')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-d0ed8f3f-641a-4a7d-a1fa-0955dae81e42 button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-d0ed8f3f-641a-4a7d-a1fa-0955dae81e42');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-73816a9e-2974-4e57-b1f1-212739d2fb47\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-73816a9e-2974-4e57-b1f1-212739d2fb47')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-73816a9e-2974-4e57-b1f1-212739d2fb47 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_689ad234-4250-41b8-a095-eee82e743eb6\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_689ad234-4250-41b8-a095-eee82e743eb6 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF ALT       QUAL  FILTER  GENOTYPE\n",
       "0      chr2    3051590 NaN   T   C  52.203800     NaN         1\n",
       "1      chr2    3051604 NaN   A   C  45.253502     NaN         1\n",
       "2      chr2    3051621 NaN   T   C  47.249500     NaN         1\n",
       "3      chr2    3051632 NaN   A   G  55.332600     NaN         1\n",
       "4      chr2    3051635 NaN   A   G  59.256699     NaN         1\n",
       "...     ...        ...  ..  ..  ..        ...     ...       ...\n",
       "70678  chr2  182002062 NaN   A   T  30.957300     NaN         1\n",
       "70679  chr2  182002065 NaN   G   A  40.932098     NaN         3\n",
       "70680  chr2  182002067 NaN   G   A  40.932098     NaN         3\n",
       "70681  chr2  182002068 NaN   A   T  30.957300     NaN         1\n",
       "70682  chr2  182002086 NaN   T   A   5.826790     NaN         1\n",
       "\n",
       "[70683 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import cyvcf2\n",
    "import pandas as pd\n",
    "\n",
    "vcf_path = 'filtered_variants_chr2.vcf'\n",
    "\n",
    "# Convert VCF to TSV\n",
    "with open('output.tsv', 'w') as file:\n",
    "    for variant in cyvcf2.VCF(vcf_path):\n",
    "        genotype = variant.gt_types[0]\n",
    "        data_string = f\"{variant.CHROM}\\t{variant.POS}\\t{variant.ID}\\t{variant.REF}\\t{','.join(variant.ALT)}\\t{variant.QUAL}\\t{variant.FILTER}\\t{genotype}\\n\"\n",
    "        file.write(data_string)\n",
    "\n",
    "# Load TSV into DataFrame\n",
    "columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'GENOTYPE']\n",
    "df = pd.read_csv(\"output.tsv\", delimiter=\"\\t\", header=None, names=columns)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 424
    },
    "id": "Y2V0TdwXk6cE",
    "outputId": "cf3e06a1-39c3-4c5f-bc61-9d02e9614d5d"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"filtered_df\",\n  \"rows\": 17761,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57461582,\n        \"min\": 3051696,\n        \"max\": 182002048,\n        \"num_unique_values\": 17759,\n        \"samples\": [\n          55775774\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 225,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 265,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54.77611585545069,\n        \"min\": 60.00130081176758,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 13556,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "filtered_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-7e49f378-ea22-4222-9bac-71461a6a6a6a\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051696</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TGTCC</td>\n",
       "      <td>96.563599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>111.313004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052386</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>129.570007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052401</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>131.856003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3052429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>84.802002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70671</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>95.972900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70672</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>71.400002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70673</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>65.766403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70675</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>77.494499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70677</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>91.032799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17761 rows × 8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7e49f378-ea22-4222-9bac-71461a6a6a6a')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-7e49f378-ea22-4222-9bac-71461a6a6a6a button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-7e49f378-ea22-4222-9bac-71461a6a6a6a');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-c8298f01-75dd-4a7a-887e-fc97c94417e8\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c8298f01-75dd-4a7a-887e-fc97c94417e8')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-c8298f01-75dd-4a7a-887e-fc97c94417e8 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_a1e6671b-8ceb-48e4-9af1-32087a1b2676\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('filtered_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_a1e6671b-8ceb-48e4-9af1-32087a1b2676 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('filtered_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF    ALT        QUAL  FILTER  GENOTYPE\n",
       "6      chr2    3051696 NaN   T  TGTCC   96.563599     NaN         1\n",
       "29     chr2    3052361 NaN   G      T  111.313004     NaN         1\n",
       "31     chr2    3052386 NaN   T      G  129.570007     NaN         1\n",
       "32     chr2    3052401 NaN   C      G  131.856003     NaN         1\n",
       "33     chr2    3052429 NaN   A      G   84.802002     NaN         1\n",
       "...     ...        ...  ..  ..    ...         ...     ...       ...\n",
       "70671  chr2  182002007 NaN   A      G   95.972900     NaN         1\n",
       "70672  chr2  182002013 NaN   T      C   71.400002     NaN         1\n",
       "70673  chr2  182002021 NaN   G      A   65.766403     NaN         1\n",
       "70675  chr2  182002039 NaN   A      G   77.494499     NaN         1\n",
       "70677  chr2  182002048 NaN   T      C   91.032799     NaN         1\n",
       "\n",
       "[17761 rows x 8 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['QUAL'] = pd.to_numeric(df['QUAL'], errors='coerce')\n",
    "\n",
    "filtered_df = df[df['QUAL'] > 60]\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 352
    },
    "id": "vZrHtbN1k__E",
    "outputId": "53116a56-0fea-4ece-a042-8f9dad8ac069"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"intended_mutations_df\",\n  \"rows\": 9,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 21704765,\n        \"min\": 36937210,\n        \"max\": 87049235,\n        \"num_unique_values\": 9,\n        \"samples\": [\n          86658391\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 6,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 51.23562662600714,\n        \"min\": 102.33000183105467,\n        \"max\": 228.25,\n        \"num_unique_values\": 9,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "intended_mutations_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-d8bec429-12bc-458a-a65c-726ca5950def\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16660</th>\n",
       "      <td>chr2</td>\n",
       "      <td>36937210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>194.453003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16667</th>\n",
       "      <td>chr2</td>\n",
       "      <td>36996899</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTTTTT</td>\n",
       "      <td>CTTTTTT</td>\n",
       "      <td>185.158005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33179</th>\n",
       "      <td>chr2</td>\n",
       "      <td>85400441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>228.171005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33346</th>\n",
       "      <td>chr2</td>\n",
       "      <td>85776687</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>115.749001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33665</th>\n",
       "      <td>chr2</td>\n",
       "      <td>85918029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>118.149002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34023</th>\n",
       "      <td>chr2</td>\n",
       "      <td>86198668</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>228.250000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34039</th>\n",
       "      <td>chr2</td>\n",
       "      <td>86236802</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CT</td>\n",
       "      <td>CTT</td>\n",
       "      <td>228.218994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34510</th>\n",
       "      <td>chr2</td>\n",
       "      <td>86658391</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>174.298996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34657</th>\n",
       "      <td>chr2</td>\n",
       "      <td>87049235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>102.330002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d8bec429-12bc-458a-a65c-726ca5950def')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-d8bec429-12bc-458a-a65c-726ca5950def button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-d8bec429-12bc-458a-a65c-726ca5950def');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-b0bffb71-1c72-4fa5-ba35-241571275877\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-b0bffb71-1c72-4fa5-ba35-241571275877')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-b0bffb71-1c72-4fa5-ba35-241571275877 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_47297c6b-5dff-4c9a-977f-284f1bfc541b\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('intended_mutations_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_47297c6b-5dff-4c9a-977f-284f1bfc541b button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('intended_mutations_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM       POS  ID     REF      ALT        QUAL  FILTER  GENOTYPE\n",
       "16660  chr2  36937210 NaN       C        T  194.453003     NaN         1\n",
       "16667  chr2  36996899 NaN  CTTTTT  CTTTTTT  185.158005     NaN         3\n",
       "33179  chr2  85400441 NaN       A        G  228.171005     NaN         3\n",
       "33346  chr2  85776687 NaN       T        A  115.749001     NaN         1\n",
       "33665  chr2  85918029 NaN       T        C  118.149002     NaN         1\n",
       "34023  chr2  86198668 NaN       A        G  228.250000     NaN         3\n",
       "34039  chr2  86236802 NaN      CT      CTT  228.218994     NaN         3\n",
       "34510  chr2  86658391 NaN       C       CG  174.298996     NaN         3\n",
       "34657  chr2  87049235 NaN       G        T  102.330002     NaN         3"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Identifying Intended Mutations:\n",
    "df = pd.read_csv(\"output.tsv\", delimiter=\"\\t\", header=None, names=columns)\n",
    "variants_df = pd.read_csv(\"output.tsv\", delimiter=\"\\t\", header=None, names=columns)\n",
    "\n",
    "intended_mutations_df = df[df['POS'].isin(intended_sites)]\n",
    "intended_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 424
    },
    "id": "IdP4Tr0MlCNK",
    "outputId": "9f54b751-2da0-4c8b-d87f-5f061b2f7246"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"off_target_mutations_df\",\n  \"rows\": 70674,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54989493,\n        \"min\": 3051590,\n        \"max\": 182002086,\n        \"num_unique_values\": 70665,\n        \"samples\": [\n          150306517\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 482,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 545,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 55.71049885114257,\n        \"min\": 3.0108699798583984,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 29707,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "off_target_mutations_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-5a9c2393-1b27-454a-950a-85d61f95a5ea\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>52.203800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>45.253502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.249500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>55.332600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>59.256699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70678</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70679</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70680</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70681</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70682</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>5.826790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70674 rows × 8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5a9c2393-1b27-454a-950a-85d61f95a5ea')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-5a9c2393-1b27-454a-950a-85d61f95a5ea button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-5a9c2393-1b27-454a-950a-85d61f95a5ea');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-ed1c1239-7aa2-45bc-aebf-bfb8d6a3c563\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ed1c1239-7aa2-45bc-aebf-bfb8d6a3c563')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-ed1c1239-7aa2-45bc-aebf-bfb8d6a3c563 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_e3ce6bcd-fac9-446b-96b2-742b0135d1a9\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('off_target_mutations_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_e3ce6bcd-fac9-446b-96b2-742b0135d1a9 button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('off_target_mutations_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF ALT       QUAL  FILTER  GENOTYPE\n",
       "0      chr2    3051590 NaN   T   C  52.203800     NaN         1\n",
       "1      chr2    3051604 NaN   A   C  45.253502     NaN         1\n",
       "2      chr2    3051621 NaN   T   C  47.249500     NaN         1\n",
       "3      chr2    3051632 NaN   A   G  55.332600     NaN         1\n",
       "4      chr2    3051635 NaN   A   G  59.256699     NaN         1\n",
       "...     ...        ...  ..  ..  ..        ...     ...       ...\n",
       "70678  chr2  182002062 NaN   A   T  30.957300     NaN         1\n",
       "70679  chr2  182002065 NaN   G   A  40.932098     NaN         3\n",
       "70680  chr2  182002067 NaN   G   A  40.932098     NaN         3\n",
       "70681  chr2  182002068 NaN   A   T  30.957300     NaN         1\n",
       "70682  chr2  182002086 NaN   T   A   5.826790     NaN         1\n",
       "\n",
       "[70674 rows x 8 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Detecting Off-Target Effects:\n",
    "off_target_mutations_df = variants_df[~variants_df['POS'].isin(intended_sites)]\n",
    "off_target_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 112
    },
    "id": "QUyPloT6lEBV",
    "outputId": "248bf3c2-1d75-4ac4-f532-73710e6df13a"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"summary_df\",\n  \"rows\": 1,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Mutation Count\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 70674,\n        \"max\": 70674,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          70674\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Genotype Count\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 70665,\n        \"max\": 70665,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          70665\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "summary_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-67f35ff9-4c44-4e85-a0f3-4ff475a51a77\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation Count</th>\n",
       "      <th>Genotype Count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr2</th>\n",
       "      <td>70674</td>\n",
       "      <td>70665</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-67f35ff9-4c44-4e85-a0f3-4ff475a51a77')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-67f35ff9-4c44-4e85-a0f3-4ff475a51a77 button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-67f35ff9-4c44-4e85-a0f3-4ff475a51a77');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "  <div id=\"id_e06f6d56-aaf0-4c0d-aad8-239d186ee68f\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('summary_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_e06f6d56-aaf0-4c0d-aad8-239d186ee68f button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('summary_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "       Mutation Count  Genotype Count\n",
       "CHROM                                \n",
       "chr2            70674           70665"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Summarizing Off-Target Mutations\n",
    "summary_df = off_target_mutations_df.groupby('CHROM')['POS'].agg(['count', 'nunique'])\n",
    "summary_df.columns = ['Mutation Count', 'Genotype Count']\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 472
    },
    "id": "ix32kdLQlG57",
    "outputId": "99b2d8dc-269d-4ae5-a01e-ed56fc13ee0d"
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAkQAAAHHCAYAAABeLEexAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAABT10lEQVR4nO3dd1QU1+M28GdpSxMQaaIICBawl4hEIxYUDVGxtyj2bxQrtpgiYolGo9glxoKxG2NJRAUbmCg2FGOPGmxRwKiIWKj3/cN35+cK6C5Snedzzh7dO3dn7p3ZXZ69e2dWIYQQICIiIpIxneJuABEREVFxYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICrlpk6dCoVCUSTbat68OZo3by7dj4qKgkKhwLZt24pk+/3794eTk1ORbCu/UlNTMXjwYNjZ2UGhUGDMmDEFuv7MzExMnDgRDg4O0NHRgZ+fX5Fsl4qGk5MT+vfvX9zNKDXCwsKgUChw+vTp4m4KfQAYiEoQ1YtbdTM0NIS9vT18fHywaNEiPH36tEC2c+/ePUydOhVxcXEFsr6CVJLbponvvvsOYWFhGDZsGNatW4e+ffu+tX5GRgYWLVqEjz76CGXKlIGpqSk++ugjLFq0CBkZGTnqr169GnPnzkXXrl2xdu1ajB079p3bVQVXTW4lzcaNG7FgwQKN6zs5OUGhUMDb2zvX5T/99JPU1/z8EX3+/DmmTp2KqKgorR+rcuzYMUydOhXJycn5Xocc7NixA+3atYOVlRUMDAxgb2+P7t2749ChQ8XdNNLQnTt3EBwcjEaNGqFs2bKwsrJC8+bNceDAgeJuWu4ElRhr1qwRAMS0adPEunXrxOrVq8V3330n2rRpIxQKhXB0dBTnzp1Te0xGRoZ48eKFVts5deqUACDWrFmj1ePS0tJEWlqadP/w4cMCgPjll1+0Wk9+25aeni5evnxZYNsqDB4eHqJJkyYa1U1NTRVeXl4CgPjss8/EkiVLxLJly0SHDh0EAOHl5SVSU1PVHtOjRw9RoUIFrbabkJAg1q1bp3arWLGiqF69eo7yksbX11c4OjpqXN/R0VEYGhoKHR0dcf/+/RzLvby8hKGhoQAgTp06pXV7Hjx4IACIoKAgrR+rMnfuXAFAxMfH51j28uVLkZ6enu91fwiys7NF//79BQBRr149MXPmTLFq1SoxY8YM0aBBAwFAHD16VAjxf++Z+TmWVPgWL14sjIyMRK9evcSSJUvEggULRP369QUAsXr16uJuXg56xZLC6K3atWuHhg0bSvcnT56MQ4cO4bPPPkOHDh1w+fJlGBkZAQD09PSgp1e4h/H58+cwNjaGgYFBoW7nXfT19Yt1+5pISkqCu7u7RnUDAwMRHR2NxYsXY8SIEVL5sGHDsHTpUowYMQLjx4/H8uXL1dZvYWGh1XZtbW3x+eefq5XNnj0bVlZWOcrzIzMzE9nZ2cX+/FBp0qQJTp06hS1btmD06NFS+d27d/HHH3+gU6dO+PXXX4uxhXlTKpXF3YRiN2/ePISFhWHMmDGYP3++2sjl119/jXXr1hXIe54QAi9fvpTeS6ngtWjRArdv34aVlZVU9sUXX6Bu3bqYMmUKBgwYUIyty0VxJzL6P+/6tPPdd98JAGLFihVSWVBQkHjzMEZGRoomTZoIc3NzYWJiIqpWrSomT54shPi/UZ03b6oRGS8vL1GjRg1x+vRp8cknnwgjIyMxevRoaZmXl5e0HdW6Nm/eLCZPnixsbW2FsbGxaN++vbh9+7ZamxwdHYW/v3+OPr2+zne1zd/fP8doQWpqqggMDBQVK1YUBgYGomrVqmLu3LkiOztbrR4AERAQIHbs2CFq1KghDAwMhLu7u9i7d2+u+/pNiYmJYuDAgcLGxkYolUpRu3ZtERYWlmNfvHnLbRRACCHu3LkjdHV1RcuWLfPcZosWLYSenp64c+eOiI+Pz3X92m5XpUaNGmrHMi0tTXz77beifv36wszMTBgbG4umTZuKQ4cOqT1O1Y65c+eKkJAQUblyZaGjoyPOnj0r7YcGDRoIpVIpKleuLEJDQ3N9jgohxLp160T9+vWFoaGhKFu2rOjRo4fa80Y1evb67V2jRY6OjsLX11f0799fNGrUSG3ZnDlzRLly5cSKFStyvM7efG6rvP6cy+sYqEaLzp07J/z9/YWzs7NQKpXC1tZWDBgwQPz333/S+lT7Iq/jldvr5MaNG6Jr166ibNmywsjISHh4eIjdu3er1VE9D7Zs2SJmzJghKlSoIJRKpWjZsqW4du2aWt2///5bdO7cWdja2gqlUikqVKggevToIZKTk9+6b4UQYuvWrdIxK1eunOjTp4+4e/dujn1mYmIi7t69Kzp27ChMTEyElZWVGDdunMjMzHzr+p8/fy4sLS1F9erV31lXiP97z/zzzz/F2LFjhZWVlTA2NhZ+fn4iKSlJra7qubFv3z7pORoSEiKE0H4fT506Vdjb2wtTU1PRpUsXkZycLF6+fClGjx4trK2thYmJiejfv3+OEe2MjAwxbdo0UblyZWFgYCAcHR3F5MmTc9Q7deqUaNOmjShXrpwwNDQUTk5OYsCAAWp1tH3v27p1q3BzcxOGhoaicePG4q+//hJCCBEaGipcXFyEUqkUXl5eub53HD9+XPj4+AgzMzNhZGQkmjVrJv788893Hp+8BAYGCgAiJSUl3+soDBwhKkX69u2Lr776CpGRkRgyZEiudS5evIjPPvsMtWvXxrRp06BUKnH9+nUcPXoUAODm5oZp06ZhypQpGDp0KD755BMAwMcffyyt4+HDh2jXrh169uyJzz//HLa2tm9t18yZM6FQKDBp0iQkJSVhwYIF8Pb2RlxcnFafvjRp2+uEEOjQoQMOHz6MQYMGoW7duoiIiMCECRPw77//IiQkRK3+n3/+ie3bt2P48OEoU6YMFi1ahC5duuD27dsoV65cnu168eIFmjdvjuvXr2PEiBFwdnbGL7/8gv79+yM5ORmjR4+Gm5sb1q1bh7Fjx6JixYoYN24cAMDa2jrXde7duxdZWVno169fntvt168fDh8+jH379qFXr15Yt24dZs6cidTUVMyaNUvaZ9psNy8pKSlYuXIlevXqhSFDhuDp06dYtWoVfHx8cPLkSdStW1et/po1a/Dy5UsMHToUSqUSlpaWOHv2LNq2bYvy5csjODgYWVlZmDZtWq5tmTlzJr799lt0794dgwcPxoMHD7B48WI0a9YMZ8+ehYWFBb7++ms8efIEd+/elY6lqampRv3p3bs32rRpgxs3bsDFxQXAq/lIXbt2zfdIo7W1NZYvX45hw4ahU6dO6Ny5MwCgdu3aAID9+/fjn3/+wYABA2BnZ4eLFy9ixYoVuHjxIo4fPw6FQoHOnTvj77//xqZNmxASEiJ9cs7reCUmJuLjjz/G8+fPMWrUKJQrVw5r165Fhw4dsG3bNnTq1Emt/uzZs6Gjo4Px48fjyZMnmDNnDvr06YMTJ04AANLT0+Hj44O0tDSMHDkSdnZ2+Pfff7F7924kJyfD3Nw8z/6HhYVhwIAB+OijjzBr1iwkJiZi4cKFOHr0qHTMVLKysuDj4wMPDw/88MMPOHDgAObNmwcXFxcMGzYsz238+eefePToEcaMGQNdXd13H5T/b+TIkShbtiyCgoJw8+ZNLFiwACNGjMCWLVvU6l29ehW9evXC//73PwwZMgTVqlXTeh/PmjULRkZG+PLLL3H9+nUsXrwY+vr60NHRwePHjzF16lQcP34cYWFhcHZ2xpQpU6THDh48GGvXrkXXrl0xbtw4nDhxArNmzcLly5exY8cOAK9Ge9u0aQNra2t8+eWXsLCwwM2bN7F9+3ZpPdq+9/3xxx/47bffEBAQIPXhs88+w8SJE7Fs2TIMHz4cjx8/xpw5czBw4EC1eVqHDh1Cu3bt0KBBAwQFBUFHRwdr1qxBy5Yt8ccff6BRo0YaHyeVhIQEGBsbw9jYWOvHFqriTmT0fzT5Ptzc3FzUq1dPuv/mp++QkBABQDx48CDPdbxtno7qU3loaGiuy3IbIapQoYJa0t+6dasAIBYuXCiVaTJC9K62vTlCtHPnTgFAzJgxQ61e165dhUKhENevX5fKAAgDAwO1snPnzgkAYvHixTm29boFCxYIAGL9+vVSWXp6uvD09BSmpqZqfVd9Cn2XMWPGCADSyEpuzpw5IwCIwMBAqUw1gvcmTber8uYIUWZmptr8MCGEePz4sbC1tRUDBw6UylSjJGZmZjk+gbdv314YGxuLf//9Vyq7du2a0NPTU3uO3rx5U+jq6oqZM2eqPf78+fNCT09PrTw/c4h8fX1FZmamsLOzE9OnTxdCCHHp0iUBQERHR+f6OtNkhEiIt88hev78eY6yTZs2CQDiyJEjUtnb5hC9+TpRPU/++OMPqezp06fC2dlZODk5iaysLCHE/70W3dzc1I7jwoULBQBx/vx5IYQQZ8+ezde8v/T0dGFjYyNq1qypNmdx9+7dAoCYMmWKVObv7y/NhXxdvXr1RIMGDd66HVV7d+zYoVG7VMfS29tbbWRk7NixQldXV23Uy9HRUQAQ+/btU1uHtvu4Zs2aavO8evXqJRQKhWjXrp3aej09PdWeO3FxcQKAGDx4sFq98ePHCwDSaOyOHTve+XdA2/c+pVKp9nz78ccfBQBhZ2en9v41efJktedmdna2qFKlivDx8VHbv8+fPxfOzs6idevWebYxL9euXROGhoaib9++Wj+2sPEss1LG1NT0rWebqT6l7dq1C9nZ2fnahlKp1Oq73X79+qFMmTLS/a5du6J8+fLYs2dPvravqT179kBXVxejRo1SKx83bhyEENi7d69aube3tzRaALz6ZG9mZoZ//vnnnduxs7NDr169pDJ9fX2MGjUKqampiI6O1rrtqmP4+n57k2pZSkqK1uvXlq6urjQHKDs7G48ePUJmZiYaNmyIM2fO5KjfpUsXtVGNrKwsHDhwAH5+frC3t5fKXV1d0a5dO7XHbt++HdnZ2ejevTv+++8/6WZnZ4cqVarg8OHDBdKf7t27Y9OmTQCADRs2wMHBQRp1LAyvj4a+fPkS//33Hxo3bgwAue5DTezZsweNGjVC06ZNpTJTU1MMHToUN2/exKVLl9TqDxgwQG0ul6q/que4agQoIiICz58/17gdp0+fRlJSEoYPHw5DQ0Op3NfXF9WrV0d4eHiOx3zxxRdq9z/55JN3vtZUz/W3vS5yM3ToULW5Rp988gmysrJw69YttXrOzs7w8fFRK9N2H/fr109tlNHDwwNCCAwcOFCtnoeHB+7cuYPMzExpO8CruYOvU43qqvah6j189+7duZ5pqlqXNu99rVq1UrtkiYeHB4BXr+PX97WqXHWc4uLicO3aNfTu3RsPHz6UXqvPnj1Dq1atcOTIEa3+zjx//hzdunWDkZERZs+erfHjigoDUSmTmpr61jeLHj16oEmTJhg8eDBsbW3Rs2dPbN26VasnbYUKFbSaIFulShW1+wqFAq6urrh586bG68iPW7duwd7ePsf+cHNzk5a/rlKlSjnWUbZsWTx+/Pid26lSpQp0dNRfLnltRxOqNr8t3GoSmvLy4MEDJCQkSLfU1NR3Pmbt2rWoXbs2DA0NUa5cOVhbWyM8PBxPnjzJUdfZ2VntflJSEl68eAFXV9ccdd8su3btGoQQqFKlCqytrdVuly9fRlJSkpa9zV3v3r1x6dIlnDt3Dhs3bkTPnj0L9dICjx49wujRo2FrawsjIyNYW1tL+ym3faiJW7duoVq1ajnKNX2Oly1bFgCk57izszMCAwOxcuVKWFlZwcfHB0uXLn1n+1Tbya0t1atXz9EOQ0PDHF8DavJaMzMzA/D210Vu3tVvlTeft8D772NVyHRwcMhRnp2dLe3bW7duQUdHJ8frwc7ODhYWFtJ2vLy80KVLFwQHB8PKygodO3bEmjVrkJaWptbm93nve1ubgf/bb9euXQMA+Pv753itrly5EmlpaRo/t7OystCzZ09cunQJ27ZtU/vgVFJwDlEpcvfuXTx58iTXPzoqRkZGOHLkCA4fPozw8HDs27cPW7ZsQcuWLREZGanR9/KFcdZFXn+IsrKytJor8D7y2o4Qoki2/zrVG9dff/2VY36Oyl9//QUAGp+19rqPPvpI7U0xKCgIU6dOzbP++vXr0b9/f/j5+WHChAmwsbGBrq4uZs2ahRs3buSo/z7PkezsbCgUCuzduzfXY6LpPKF38fDwgIuLC8aMGYP4+Hj07t07z7oKhSLX50FWVpbG2+vevTuOHTuGCRMmoG7dujA1NUV2djbatm2b79FabWnyHJ83bx769++PXbt2ITIyEqNGjcKsWbNw/PhxVKxYsVDb8S7Vq1cHAJw/f1666Oj7bO/NY1oQ7215bUvTNrwrlKsudnv8+HH8/vvviIiIwMCBAzFv3jwcP348X6+P/LZZ9bydO3dunu9TmrZnyJAh2L17NzZs2ICWLVtq9JiixkBUiqxbtw4Acgz5vklHRwetWrVCq1atMH/+fHz33Xf4+uuvcfjwYXh7exf4p2TVpwgVIQSuX78uTTYFXn1iy+1CdLdu3ULlypWl+9q0zdHREQcOHMDTp0/VPilduXJFWl4QHB0d8ddffyE7O1ttlOh9ttOuXTvo6upi3bp1eU6s/vnnn6Gnp4e2bdtqvf4NGzbgxYsX0v3X93Futm3bhsqVK2P79u1qxyAoKEij7dnY2MDQ0BDXr1/PsezNMhcXFwgh4OzsjKpVq751ve/7XO3VqxdmzJgBNze3PN/QgVfPz9y+znnzk3Ze7Xn8+DEOHjyI4OBgtUm0b7423raO3Dg6OuLq1as5yt/3OV6rVi3UqlUL33zzDY4dO4YmTZogNDQUM2bMyLMdwKtJyW/+Mbt69WqBvdaaNm2KsmXLYtOmTfjqq6+K5MNSYe3j3LaTnZ2Na9euSR+IgFcT55OTk3Nsp3HjxmjcuDFmzpyJjRs3ok+fPti8eTMGDx5cZO99qikGZmZmeV7sVBMTJkzAmjVrsGDBArWpByUNvzIrJQ4dOoTp06fD2dkZffr0ybPeo0ePcpSp/hCohlxNTEwAoMCulPvzzz+rDXFv27YN9+/fV5s74uLiguPHjyM9PV0q2717N+7cuaO2Lm3a9umnnyIrKwtLlixRKw8JCYFCocgxdyW/Pv30UyQkJKidsZKZmYnFixfD1NQUXl5eWq/TwcEBAwYMwIEDB9SuM6QSGhqKQ4cOYdCgQfn61N6kSRN4e3tLt3cFItUfntc/zZ44cQIxMTEabU9XVxfe3t7YuXMn7t27J5Vfv349x3yGzp07Q1dXF8HBwTk+PQsh8PDhQ+m+iYlJvr9uAl6d1RMUFIR58+a9tZ6LiwuuXLmCBw8eSGXnzp2Tzs5UUZ0V8+bzM7f9ByDXq2xr+xw/efKk2nF49uwZVqxYAScnJ61HD1NSUqQ5LSq1atWCjo6O2lcyb2rYsCFsbGwQGhqqVm/v3r24fPkyfH19tWpHXoyNjTFp0iRcvnwZkyZNynXUbv369Th58mSBbA8o+H38tu0AOZ8T8+fPBwBpHz5+/DhHv998Dy+q974GDRrAxcUFP/zwQ65fu7/+esnL3Llz8cMPP+Crr75Suy5YScQRohJo7969uHLlCjIzM5GYmIhDhw5h//79cHR0xG+//aY2qfFN06ZNw5EjR+Dr6wtHR0ckJSVh2bJlqFixojRp0MXFBRYWFggNDUWZMmVgYmICDw+PXL9f14SlpSWaNm2KAQMGIDExEQsWLICrq6vapQEGDx6Mbdu2oW3btujevTtu3LiB9evXq01y1rZt7du3R4sWLfD111/j5s2bqFOnDiIjI7Fr1y6MGTMmx7rza+jQofjxxx/Rv39/xMbGwsnJCdu2bcPRo0exYMGCfM3xAV69eV25cgXDhw/Hvn37pJGgiIgI7Nq1C15eXu/8Q15QPvvsM2zfvh2dOnWCr68v4uPjERoaCnd3d43mHwGvflcvMjISTZo0wbBhw6Q37Jo1a6r9FIuLiwtmzJiByZMn4+bNm/Dz80OZMmUQHx+PHTt2YOjQoRg/fjyAV2/IW7ZsQWBgID766COYmpqiffv2GvfL0dHxrV8VqgwcOBDz58+Hj48PBg0ahKSkJISGhqJGjRpqk9qNjIzg7u6OLVu2oGrVqrC0tETNmjVRs2ZNNGvWDHPmzEFGRgYqVKiAyMhIxMfH59hWgwYNALy6yGDPnj2hr6+P9u3bS0HpdV9++SU2bdqEdu3aYdSoUbC0tMTatWsRHx+PX3/9Nce8tnc5dOgQRowYgW7duqFq1arIzMzEunXroKuriy5duuT5OH19fXz//fcYMGAAvLy80KtXL+m0eycnJ+knZArChAkTcPHiRcybNw+HDx9G165dYWdnh4SEBOzcuRMnT57EsWPHCmx7Bb2P81KnTh34+/tjxYoVSE5OhpeXF06ePIm1a9fCz88PLVq0APBqLt+yZcvQqVMnuLi44OnTp/jpp59gZmYmhaqieu/T0dHBypUr0a5dO9SoUQMDBgxAhQoV8O+//+Lw4cMwMzPD77//nufjd+zYgYkTJ6JKlSpwc3PD+vXr1Za3bt36nZd1KVJFfVob5U11CqnqZmBgIOzs7ETr1q3FwoULc72I1Zun3R88eFB07NhR2NvbCwMDA2Fvby969eol/v77b7XH7dq1S7i7u0unRL95Ycbc5HXa/aZNm8TkyZOFjY2NMDIyEr6+vuLWrVs5Hj9v3jzpgnFNmjQRp0+fzvV057zaltuFGZ8+fSrGjh0r7O3thb6+vqhSpcpbL072prwuB/CmxMREMWDAAGFlZSUMDAxErVq1cr00gLanv6elpYmQkBDRoEEDYWJiIoyNjUX9+vXFggULcv0Jh8I67T47O1t89913wtHRUSiVSlGvXj2xe/fuHPv89Qsz5ubgwYOiXr16wsDAQLi4uIiVK1eKcePGCUNDwxx1f/31V9G0aVNhYmIiTExMRPXq1UVAQIC4evWqVCc1NVX07t1bWFhYaHVhxrfJ6/IW69evly6YV7duXREREZHrc+7YsWOiQYMGwsDAQO0U/Lt374pOnToJCwsLYW5uLrp16ybu3buX62n606dPFxUqVBA6OjoaX5jRwsJCGBoaikaNGuV50cA3T6dXHS/Vc/Wff/4RAwcOFC4uLsLQ0FBYWlqKFi1aiAMHDrx1n6ls2bJF1KtXTyiVSmFpafnWCzO+Ka8LdOZl27Ztok2bNsLS0lLo6emJ8uXLix49eoioqCipTl7HUrU/Dh8+LJW97bnxPvs4rzao+vv6JVAyMjJEcHCwcHZ2Fvr6+sLBwSHHhRnPnDkjevXqJSpVqiSUSqWwsbERn332mTh9+rTa+t/nvS+v13FefTx79qzo3LmzKFeunFAqlcLR0VF0795dHDx4MNf9+eY+yOv2+vEpCRRCFMOMUiKSDT8/P1y8eDHX+TRERCUF5xARUYF5fSI38GpS8Z49e9C8efPiaRARkYY4QkREBaZ8+fLo378/KleujFu3bmH58uVIS0vD2bNnc1yvioioJOGkaiIqMG3btsWmTZuQkJAApVIJT09PfPfddwxDRFTicYSIiIiIZI9ziIiIiEj2GIiIiIhI9jiHSAPZ2dm4d+8eypQpU6g/DklEREQFRwiBp0+fwt7e/p0X2WQg0sC9e/dy/CowERERlQ537tx5588gMRBpQPXTDHfu3IGZmVkxt4aIiIg0kZKSAgcHB41+YomBSAOqr8nMzMwYiIiIiEoZTaa7cFI1ERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJnl5xN4CIiIg+bE5fhr+zzs3ZvkXQkrxxhIiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSvxASi2bNnQ6FQYMyYMVLZy5cvERAQgHLlysHU1BRdunRBYmKi2uNu374NX19fGBsbw8bGBhMmTEBmZqZanaioKNSvXx9KpRKurq4ICwsrgh4RERFRaVEiAtGpU6fw448/onbt2mrlY8eOxe+//45ffvkF0dHRuHfvHjp37iwtz8rKgq+vL9LT03Hs2DGsXbsWYWFhmDJlilQnPj4evr6+aNGiBeLi4jBmzBgMHjwYERERRdY/IiIiKtmKPRClpqaiT58++Omnn1C2bFmp/MmTJ1i1ahXmz5+Pli1bokGDBlizZg2OHTuG48ePAwAiIyNx6dIlrF+/HnXr1kW7du0wffp0LF26FOnp6QCA0NBQODs7Y968eXBzc8OIESPQtWtXhISEFEt/iYiIqOQp9kAUEBAAX19feHt7q5XHxsYiIyNDrbx69eqoVKkSYmJiAAAxMTGoVasWbG1tpTo+Pj5ISUnBxYsXpTpvrtvHx0daBxEREZFecW588+bNOHPmDE6dOpVjWUJCAgwMDGBhYaFWbmtri4SEBKnO62FItVy17G11UlJS8OLFCxgZGeXYdlpaGtLS0qT7KSkp2neOiIiISo1iGyG6c+cORo8ejQ0bNsDQ0LC4mpGrWbNmwdzcXLo5ODgUd5OIiIioEBVbIIqNjUVSUhLq168PPT096OnpITo6GosWLYKenh5sbW2Rnp6O5ORktcclJibCzs4OAGBnZ5fjrDPV/XfVMTMzy3V0CAAmT56MJ0+eSLc7d+4URJeJiIiohCq2QNSqVSucP38ecXFx0q1hw4bo06eP9H99fX0cPHhQeszVq1dx+/ZteHp6AgA8PT1x/vx5JCUlSXX2798PMzMzuLu7S3VeX4eqjmoduVEqlTAzM1O7ERER0Yer2OYQlSlTBjVr1lQrMzExQbly5aTyQYMGITAwEJaWljAzM8PIkSPh6emJxo0bAwDatGkDd3d39O3bF3PmzEFCQgK++eYbBAQEQKlUAgC++OILLFmyBBMnTsTAgQNx6NAhbN26FeHh4UXbYSIiIiqxinVS9buEhIRAR0cHXbp0QVpaGnx8fLBs2TJpua6uLnbv3o1hw4bB09MTJiYm8Pf3x7Rp06Q6zs7OCA8Px9ixY7Fw4UJUrFgRK1euhI+PT3F0iYiIiEoghRBCFHcjSrqUlBSYm5vjyZMn/PqMiIhIS05fvvtbmZuzfQt8u9r8/S726xARERERFTcGIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPQYiIiIikj0GIiIiIpI9BiIiIiKSPb3ibgARUX45fRn+zjo3Z/sWQUuIqLTjCBERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJXrEGouXLl6N27dowMzODmZkZPD09sXfvXmn5y5cvERAQgHLlysHU1BRdunRBYmKi2jpu374NX19fGBsbw8bGBhMmTEBmZqZanaioKNSvXx9KpRKurq4ICwsriu4RERFRKVGsgahixYqYPXs2YmNjcfr0abRs2RIdO3bExYsXAQBjx47F77//jl9++QXR0dG4d+8eOnfuLD0+KysLvr6+SE9Px7Fjx7B27VqEhYVhypQpUp34+Hj4+vqiRYsWiIuLw5gxYzB48GBEREQUeX+JiIioZFIIIYQ2D2jZsiW2b98OCwsLtfKUlBT4+fnh0KFD79UgS0tLzJ07F127doW1tTU2btyIrl27AgCuXLkCNzc3xMTEoHHjxti7dy8+++wz3Lt3D7a2tgCA0NBQTJo0CQ8ePICBgQEmTZqE8PBwXLhwQdpGz549kZycjH379mnUppSUFJibm+PJkycwMzN7r/59KJy+DH9nnZuzfYugJSRnfB4SlQ7F9VrV5u+31iNEUVFRSE9Pz1H+8uVL/PHHH9quTpKVlYXNmzfj2bNn8PT0RGxsLDIyMuDt7S3VqV69OipVqoSYmBgAQExMDGrVqiWFIQDw8fFBSkqKNMoUExOjtg5VHdU6cpOWloaUlBS1GxEREX249DSt+Ndff0n/v3TpEhISEqT7WVlZ2LdvHypUqKB1A86fPw9PT0+8fPkSpqam2LFjB9zd3REXFwcDA4McI1G2trbSthMSEtTCkGq5atnb6qSkpODFixcwMjLK0aZZs2YhODhY674QERFR6aRxIKpbty4UCgUUCgVatmyZY7mRkREWL16sdQOqVauGuLg4PHnyBNu2bYO/vz+io6O1Xk9Bmjx5MgIDA6X7KSkpcHBwKMYWERERUWHSOBDFx8dDCIHKlSvj5MmTsLa2lpYZGBjAxsYGurq6WjfAwMAArq6uAIAGDRrg1KlTWLhwIXr06IH09HQkJyerjRIlJibCzs4OAGBnZ4eTJ0+qrU91Ftrrdd48My0xMRFmZma5jg4BgFKphFKp1LovREREVDppHIgcHR0BANnZ2YXWGNX609LS0KBBA+jr6+PgwYPo0qULAODq1au4ffs2PD09AQCenp6YOXMmkpKSYGNjAwDYv38/zMzM4O7uLtXZs2eP2jb2798vrYOIiIhI40D0umvXruHw4cNISkrKEZBeP+X9XSZPnox27dqhUqVKePr0KTZu3IioqChERETA3NwcgwYNQmBgICwtLWFmZoaRI0fC09MTjRs3BgC0adMG7u7u6Nu3L+bMmYOEhAR88803CAgIkEZ4vvjiCyxZsgQTJ07EwIEDcejQIWzduhXh4e+e8U5ERETyoHUg+umnnzBs2DBYWVnBzs4OCoVCWqZQKLQKRElJSejXrx/u378Pc3Nz1K5dGxEREWjdujUAICQkBDo6OujSpQvS0tLg4+ODZcuWSY/X1dXF7t27MWzYMHh6esLExAT+/v6YNm2aVMfZ2Rnh4eEYO3YsFi5ciIoVK2LlypXw8fHRtutERES54iUgSj+tA9GMGTMwc+ZMTJo06b03vmrVqrcuNzQ0xNKlS7F06dI86zg6Oub4SuxNzZs3x9mzZ/PVRiIiIvrwaX0dosePH6Nbt26F0RYiIiKiYqF1IOrWrRsiIyMLoy1ERERExULrr8xcXV3x7bff4vjx46hVqxb09fXVlo8aNarAGkdERERUFLQORCtWrICpqSmio6NzXEBRoVAwEBEREVGpo3Ugio+PL4x2EBERERUbrecQEREREX1otB4hGjhw4FuXr169Ot+NISIiIioOWgeix48fq93PyMjAhQsXkJycnOuPvhIREREv3ljSaR2IduzYkaMsOzsbw4YNg4uLS4E0ioiIiKgoFcgcIh0dHQQGBiIkJKQgVkdERERUpApsUvWNGzeQmZlZUKsjIiIiKjJaf2UWGBiodl8Igfv37yM8PBz+/v4F1jAiIiKioqJ1IHrzR1J1dHRgbW2NefPmvfMMNCIiIqKSSOtAdPjw4cJoBxEREVGx0ToQqTx48ABXr14FAFSrVg3W1tYF1igiIiKioqT1pOpnz55h4MCBKF++PJo1a4ZmzZrB3t4egwYNwvPnzwujjURERESFSutAFBgYiOjoaPz+++9ITk5GcnIydu3ahejoaIwbN64w2khERERUqLT+yuzXX3/Ftm3b0Lx5c6ns008/hZGREbp3747ly5cXZPuIiIiICp3WI0TPnz+Hra1tjnIbGxt+ZUZERESlktaByNPTE0FBQXj58qVU9uLFCwQHB8PT07NAG0dERERUFLT+ymzhwoXw8fFBxYoVUadOHQDAuXPnYGhoiIiIiAJvIBEREVFh0zoQ1axZE9euXcOGDRtw5coVAECvXr3Qp08fGBkZFXgDiYiIiApbvq5DZGxsjCFDhhR0W4iIiIiKhcZziGJjY9GiRQukpKTkWPbkyRO0aNEC586dK9DGERERERUFjQPRvHnz0LJlS5iZmeVYZm5ujtatW2Pu3LkF2jgiIiKioqBxIDpx4gQ6duyY5/L27dvj2LFjBdIoIiIioqKkcSD6999/UaZMmTyXm5qa4v79+wXSKCIiIqKipHEgsra2ln7MNTdXrlyBlZVVgTSKiIiIqChpHIi8vb0xc+bMXJcJITBz5kx4e3sXWMOIiIiIiorGp91/8803aNCgATw8PDBu3DhUq1YNwKuRoXnz5uHvv/9GWFhYYbWTiIiIqNBoHIhcXFxw4MAB9O/fHz179oRCoQDwanTI3d0d+/fvh6ura6E1lIiIiKiwaHVhxoYNG+LChQuIi4vDtWvXIIRA1apVUbdu3UJqHhEREVHhy9eVquvWrcsQRERERB8MrX/tnoiIiOhDw0BEREREssdARERERLLHQERERESyl69J1cnJyTh58iSSkpKQnZ2ttqxfv34F0jAiIiKioqJ1IPr999/Rp08fpKamwszMTLoeEQAoFAoGIiIiIip1tP7KbNy4cRg4cCBSU1ORnJyMx48fS7dHjx4VRhuJiIiICpXWgejff//FqFGjYGxsXBjtISIiIipyWgciHx8fnD59ujDaQkRERFQstJ5D5OvriwkTJuDSpUuoVasW9PX11ZZ36NChwBpHREREVBS0DkRDhgwBAEybNi3HMoVCgaysrPdvFREREVER0joQvXmaPX14nL4ML+4mEBERFSlemJGIiIhkL1+BKDo6Gu3bt4erqytcXV3RoUMH/PHHHwXdNiIiIqIioXUgWr9+Pby9vWFsbIxRo0Zh1KhRMDIyQqtWrbBx48bCaCMRERFRodJ6DtHMmTMxZ84cjB07ViobNWoU5s+fj+nTp6N3794F2kAiIiKiwqb1CNE///yD9u3b5yjv0KED4uPjC6RRREREREVJ60Dk4OCAgwcP5ig/cOAAHBwcCqRRREREREVJ66/Mxo0bh1GjRiEuLg4ff/wxAODo0aMICwvDwoULC7yBRERERIVN60A0bNgw2NnZYd68edi6dSsAwM3NDVu2bEHHjh0LvIFEREREhU3rQAQAnTp1QqdOnQq6LURERETFIl+BiIiIKD80uRL+zdm+RdASInUaBSJLS0v8/fffsLKyQtmyZaFQKPKs++jRowJrHBEREVFR0CgQhYSEoEyZMtL/3xaIiIiIiEobjQKRv7+/9P/+/fsXVluIChWH6omIKC9aX4dIV1cXSUlJOcofPnwIXV3dAmkUERERUVHSOhAJIXItT0tLg4GBwXs3iIiIiKioaXyW2aJFiwAACoUCK1euhKmpqbQsKysLR44cQfXq1Qu+hURERESFTONAFBISAuDVCFFoaKja12MGBgZwcnJCaGhowbeQiIiIqJBpHIhUP9zaokULbN++HWXLli20RhEREREVJa0vzHj48OHCaAcRERFRscnXlarv3r2L3377Dbdv30Z6errasvnz5xdIw4iIiIiKitaB6ODBg+jQoQMqV66MK1euoGbNmrh58yaEEKhfv35htJGIiIioUGl92v3kyZMxfvx4nD9/HoaGhvj1119x584deHl5oVu3boXRRiIiIqJCpXUgunz5Mvr16wcA0NPTw4sXL2Bqaopp06bh+++/12pds2bNwkcffYQyZcrAxsYGfn5+uHr1qlqdly9fIiAgAOXKlYOpqSm6dOmCxMREtTq3b9+Gr68vjI2NYWNjgwkTJiAzM1OtTlRUFOrXrw+lUglXV1eEhYVp23UiKiBOX4a/80ZEVJS0DkQmJibSvKHy5cvjxo0b0rL//vtPq3VFR0cjICAAx48fx/79+5GRkYE2bdrg2bNnUp2xY8fi999/xy+//ILo6Gjcu3cPnTt3lpZnZWXB19cX6enpOHbsGNauXYuwsDBMmTJFqhMfHw9fX1+0aNECcXFxGDNmDAYPHoyIiAhtu09EREQfIK3nEDVu3Bh//vkn3Nzc8Omnn2LcuHE4f/48tm/fjsaNG2u1rn379qndDwsLg42NDWJjY9GsWTM8efIEq1atwsaNG9GyZUsAwJo1a+Dm5objx4+jcePGiIyMxKVLl3DgwAHY2tqibt26mD59OiZNmoSpU6fCwMAAoaGhcHZ2xrx58wAAbm5u+PPPPxESEgIfHx9tdwFpiL8dRkREpYXWI0Tz58+Hh4cHACA4OBitWrXCli1b4OTkhFWrVr1XY548eQIAsLS0BADExsYiIyMD3t7eUp3q1aujUqVKiImJAQDExMSgVq1asLW1ler4+PggJSUFFy9elOq8vg5VHdU63pSWloaUlBS1GxEREX24tB4hqly5svR/ExOTArs6dXZ2NsaMGYMmTZqgZs2aAICEhAQYGBjAwsJCra6trS0SEhKkOq+HIdVy1bK31UlJScGLFy9gZGSktmzWrFkIDg4ukH4RERFRyaf1CFHlypXx8OHDHOXJyclqYUlbAQEBuHDhAjZv3pzvdRSUyZMn48mTJ9Ltzp07xd0kIiIiKkRajxDdvHkTWVlZOcrT0tLw77//5qsRI0aMwO7du3HkyBFUrFhRKrezs0N6ejqSk5PVRokSExNhZ2cn1Tl58qTa+lRnob1e580z0xITE2FmZpZjdAgAlEollEplvvpCREREpY/Ggei3336T/h8REQFzc3PpflZWFg4ePAgnJyetNi6EwMiRI7Fjxw5ERUXB2dlZbXmDBg2gr6+PgwcPokuXLgCAq1ev4vbt2/D09AQAeHp6YubMmUhKSoKNjQ0AYP/+/TAzM4O7u7tUZ8+ePWrr3r9/v7QOIiIikjeNA5Gfnx8AQKFQwN/fX22Zvr4+nJycpLO4NBUQEICNGzdi165dKFOmjDTnx9zcHEZGRjA3N8egQYMQGBgIS0tLmJmZYeTIkfD09JTOaGvTpg3c3d3Rt29fzJkzBwkJCfjmm28QEBAgjfJ88cUXWLJkCSZOnIiBAwfi0KFD2Lp1K8LDea2T4sYz0YiIqCTQOBBlZ2cDAJydnXHq1ClYWVm998aXL18OAGjevLla+Zo1a9C/f38AQEhICHR0dNClSxekpaXBx8cHy5Ytk+rq6upi9+7dGDZsGDw9PWFiYgJ/f39MmzZNquPs7Izw8HCMHTsWCxcuRMWKFbFy5Uqeck9EREQA8jGHKD4+vsA2LoR4Zx1DQ0MsXboUS5cuzbOOo6Njjq/E3tS8eXOcPXtW6zYSERHRh0/rQPT6yEtuXr9CNBEREVFpoHUg2rFjh9r9jIwMxMfHQ09PDy4uLgxEREREVOpoHYhy+9opJSUF/fv3R6dOnQqkUURERERFSesLM+bGzMwMwcHB+PbbbwtidURERERFqkACEQDpqs5EREREpY3WX5ktWrRI7b4QAvfv38e6devQrl27AmsYERERUVHROhCFhISo3dfR0YG1tTX8/f0xefLkAmsYkTY0ucAjERFRXor1OkREREREJUGBzSEiIiIiKq00HiEaOHCgRvVWr16d78bIFX/Pi4iIqHhpHIjCwsLg6OiIevXqafSTG0RERESlhcaBaNiwYdi0aRPi4+MxYMAAfP7557C0tCzMthEREREVCY0D0dKlSzF//nxs374dq1evxuTJk+Hr64tBgwahTZs2UCgUhdlOKiA8G4uIiCgnrSZVK5VK9OrVC/v378elS5dQo0YNDB8+HE5OTkhNTS2sNhIREREVqnyfZaajowOFQgEhBLKysgqyTURERERFSqtAlJaWhk2bNqF169aoWrUqzp8/jyVLluD27dswNTUtrDYSERERFSqN5xANHz4cmzdvhoODAwYOHIhNmzbBysqqMNtGREREVCQ0DkShoaGoVKkSKleujOjoaERHR+dab/v27QXWOCIiIqKioHEg6tevH88kIyIiog+SVhdmJCIiIvoQ8bfMiIiISPYYiIiIiEj2GIiIiIhI9hiIiIiISPYYiIiIiEj2GIiIiIhI9hiIiIiISPYYiIiIiEj2GIiIiIhI9hiIiIiISPY0/ukOIiIqOZy+DNeo3s3ZvoXcEqIPA0eIiIiISPY4QkRERPQWmo7GUenGQEREJRL/CBFRUeJXZkRERCR7DEREREQkewxEREREJHucQ0RERPSB0WQOHi/JoI4jRERERCR7DEREREQkewxEREREJHsMRERERCR7DEREREQkewxEREREJHsMRERERCR7DEREREQkewxEREREJHsMRERERCR7DEREREQke/wtMyLSGH8fiYg+VBwhIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZ41lmRPRB45lxRKQJjhARERGR7HGEiIiIZEuTEUSSB44QERERkewxEBEREZHsMRARERGR7HEOEVEh4JlNRESlC0eIiIiISPY4QkRERES5ktNoNwMREQHg6cdyJqc/ekR5YSAiKsH4h4rkiM97Kg6cQ0RERESyV6wjREeOHMHcuXMRGxuL+/fvY8eOHfDz85OWCyEQFBSEn376CcnJyWjSpAmWL1+OKlWqSHUePXqEkSNH4vfff4eOjg66dOmChQsXwtTUVKrz119/ISAgAKdOnYK1tTVGjhyJiRMnFmVXiYg0xq8vCwb3I2mjWEeInj17hjp16mDp0qW5Lp8zZw4WLVqE0NBQnDhxAiYmJvDx8cHLly+lOn369MHFixexf/9+7N69G0eOHMHQoUOl5SkpKWjTpg0cHR0RGxuLuXPnYurUqVixYkWh94+IiIhKh2IdIWrXrh3atWuX6zIhBBYsWIBvvvkGHTt2BAD8/PPPsLW1xc6dO9GzZ09cvnwZ+/btw6lTp9CwYUMAwOLFi/Hpp5/ihx9+gL29PTZs2ID09HSsXr0aBgYGqFGjBuLi4jB//ny14ERERO+HIzJUmpXYSdXx8fFISEiAt7e3VGZubg4PDw/ExMSgZ8+eiImJgYWFhRSGAMDb2xs6Ojo4ceIEOnXqhJiYGDRr1gwGBgZSHR8fH3z//fd4/PgxypYtm2PbaWlpSEtLk+6npKQUUi9JzvjHg4jyg+8dhaPEBqKEhAQAgK2trVq5ra2ttCwhIQE2NjZqy/X09GBpaalWx9nZOcc6VMtyC0SzZs1CcHBwwXSkCPFFQkRElD8lNhAVp8mTJyMwMFC6n5KSAgcHh2JsERUVnu5LRCRPJTYQ2dnZAQASExNRvnx5qTwxMRF169aV6iQlJak9LjMzE48ePZIeb2dnh8TERLU6qvuqOm9SKpVQKpUF0g+ikoCjh0REb1dir0Pk7OwMOzs7HDx4UCpLSUnBiRMn4OnpCQDw9PREcnIyYmNjpTqHDh1CdnY2PDw8pDpHjhxBRkaGVGf//v2oVq1arl+XERERkfwU6whRamoqrl+/Lt2Pj49HXFwcLC0tUalSJYwZMwYzZsxAlSpV4OzsjG+//Rb29vbStYrc3NzQtm1bDBkyBKGhocjIyMCIESPQs2dP2NvbAwB69+6N4OBgDBo0CJMmTcKFCxewcOFChISEFEeXiT54HI16O+4fopKpWAPR6dOn0aJFC+m+at6Ov78/wsLCMHHiRDx79gxDhw5FcnIymjZtin379sHQ0FB6zIYNGzBixAi0atVKujDjokWLpOXm5uaIjIxEQEAAGjRoACsrK0yZMoWn3Jci/ANCRESFrVgDUfPmzSGEyHO5QqHAtGnTMG3atDzrWFpaYuPGjW/dTu3atfHHH3/ku51ERET0YSuxc4iIiIiIigoDEREREcleiT3tnoioJOFcNqIPGwMRERG9EwMhfegYiEoJvhkREREVHs4hIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItnjpGoikj2etEBEHCEiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZ42n3REREMsTLTahjICIiIiohGFKKDwMRkZb4hkVE9OFhICIi+oB9qAH+Q+0XFR8GIqJSjn8YiIjeH88yIyIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZYyAiIiIi2WMgIiIiItljICIiIiLZ0yvuBhAREVHp5fRleHE3oUBwhIiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkT1aBaOnSpXBycoKhoSE8PDxw8uTJ4m4SERERlQCyCURbtmxBYGAggoKCcObMGdSpUwc+Pj5ISkoq7qYRERFRMZNNIJo/fz6GDBmCAQMGwN3dHaGhoTA2Nsbq1auLu2lERERUzGQRiNLT0xEbGwtvb2+pTEdHB97e3oiJiSnGlhEREVFJIIvfMvvvv/+QlZUFW1tbtXJbW1tcuXIlR/20tDSkpaVJ9588eQIASElJKZT2Zac9L5T1EhERlRaF8TdWtU4hxDvryiIQaWvWrFkIDg7OUe7g4FAMrSEiIvrwmS8ovHU/ffoU5ubmb60ji0BkZWUFXV1dJCYmqpUnJibCzs4uR/3JkycjMDBQup+dnY1Hjx6hXLlyUCgUWm07JSUFDg4OuHPnDszMzPLXgVKKfWff2Xf5YN/Z95LYdyEEnj59Cnt7+3fWlUUgMjAwQIMGDXDw4EH4+fkBeBVyDh48iBEjRuSor1QqoVQq1cosLCzeqw1mZmYl8slSFNh39l1u2Hf2XW5Kct/fNTKkIotABACBgYHw9/dHw4YN0ahRIyxYsADPnj3DgAEDirtpREREVMxkE4h69OiBBw8eYMqUKUhISEDdunWxb9++HBOtiYiISH5kE4gAYMSIEbl+RVaYlEolgoKCcnwFJwfsO/suN+w7+y43H1LfFUKTc9GIiIiIPmCyuDAjERER0dswEBEREZHsMRARERGR7DEQERERkewxEOXD0qVL4eTkBENDQ3h4eODkyZNvrf/LL7+gevXqMDQ0RK1atbBnzx615UIITJkyBeXLl4eRkRG8vb1x7dq1wuxCvmnT959++gmffPIJypYti7Jly8Lb2ztH/f79+0OhUKjd2rZtW9jdyBdt+h4WFpajX4aGhmp1PtTj3rx58xx9VygU8PX1leqUhuN+5MgRtG/fHvb29lAoFNi5c+c7HxMVFYX69etDqVTC1dUVYWFhOepo+/5RHLTt+/bt29G6dWtYW1vDzMwMnp6eiIiIUKszderUHMe8evXqhdiL/NG271FRUbk+3xMSEtTqfYjHPbfXsUKhQI0aNaQ6peW4AwxEWtuyZQsCAwMRFBSEM2fOoE6dOvDx8UFSUlKu9Y8dO4ZevXph0KBBOHv2LPz8/ODn54cLFy5IdebMmYNFixYhNDQUJ06cgImJCXx8fPDy5cui6pZGtO17VFQUevXqhcOHDyMmJgYODg5o06YN/v33X7V6bdu2xf3796Xbpk2biqI7WtG278CrK7e+3q9bt26pLf9Qj/v27dvV+n3hwgXo6uqiW7duavVK+nF/9uwZ6tSpg6VLl2pUPz4+Hr6+vmjRogXi4uIwZswYDB48WC0Y5Od5VBy07fuRI0fQunVr7NmzB7GxsWjRogXat2+Ps2fPqtWrUaOG2jH/888/C6P570XbvqtcvXpVrW82NjbSsg/1uC9cuFCtz3fu3IGlpWWO13ppOO4AAEFaadSokQgICJDuZ2VlCXt7ezFr1qxc63fv3l34+vqqlXl4eIj//e9/QgghsrOzhZ2dnZg7d660PDk5WSiVSrFp06ZC6EH+adv3N2VmZooyZcqItWvXSmX+/v6iY8eOBd3UAqdt39esWSPMzc3zXJ+cjntISIgoU6aMSE1NlcpKy3FXASB27Njx1joTJ04UNWrUUCvr0aOH8PHxke6/774sDpr0PTfu7u4iODhYuh8UFCTq1KlTcA0rApr0/fDhwwKAePz4cZ515HLcd+zYIRQKhbh586ZUVpqOO0eItJCeno7Y2Fh4e3tLZTo6OvD29kZMTEyuj4mJiVGrDwA+Pj5S/fj4eCQkJKjVMTc3h4eHR57rLA756fubnj9/joyMDFhaWqqVR0VFwcbGBtWqVcOwYcPw8OHDAm37+8pv31NTU+Ho6AgHBwd07NgRFy9elJbJ6bivWrUKPXv2hImJiVp5ST/u2nrXa70g9mVpkZ2djadPn+Z4rV+7dg329vaoXLky+vTpg9u3bxdTCwte3bp1Ub58ebRu3RpHjx6VyuV03FetWgVvb284OjqqlZeW485ApIX//vsPWVlZOX7uw9bWNsf3xSoJCQlvra/6V5t1Fof89P1NkyZNgr29vdobQ9u2bfHzzz/j4MGD+P777xEdHY127dohKyurQNv/PvLT92rVqmH16tXYtWsX1q9fj+zsbHz88ce4e/cuAPkc95MnT+LChQsYPHiwWnlpOO7ayuu1npKSghcvXhTIa6i0+OGHH5Camoru3btLZR4eHggLC8O+ffuwfPlyxMfH45NPPsHTp0+LsaXvr3z58ggNDcWvv/6KX3/9FQ4ODmjevDnOnDkDoGDeO0uDe/fuYe/evTle66XpuMvqpzuo+MyePRubN29GVFSU2uTinj17Sv+vVasWateuDRcXF0RFRaFVq1bF0dQC4enpCU9PT+n+xx9/DDc3N/z444+YPn16MbasaK1atQq1atVCo0aN1Mo/1ONOwMaNGxEcHIxdu3apzaNp166d9P/atWvDw8MDjo6O2Lp1KwYNGlQcTS0Q1apVQ7Vq1aT7H3/8MW7cuIGQkBCsW7euGFtWtNauXQsLCwv4+fmplZem484RIi1YWVlBV1cXiYmJauWJiYmws7PL9TF2dnZvra/6V5t1Fof89F3lhx9+wOzZsxEZGYnatWu/tW7lypVhZWWF69evv3ebC8r79F1FX18f9erVk/olh+P+7NkzbN68WaM3vZJ43LWV12vdzMwMRkZGBfI8Kuk2b96MwYMHY+vWrTm+PnyThYUFqlatWqqPeV4aNWok9UsOx10IgdWrV6Nv374wMDB4a92SfNwZiLRgYGCABg0a4ODBg1JZdnY2Dh48qDYa8DpPT0+1+gCwf/9+qb6zszPs7OzU6qSkpODEiRN5rrM45KfvwKszqaZPn459+/ahYcOG79zO3bt38fDhQ5QvX75A2l0Q8tv312VlZeH8+fNSvz704w68utxEWloaPv/883dupyQed22967VeEM+jkmzTpk0YMGAANm3apHaJhbykpqbixo0bpfqY5yUuLk7q14d+3AEgOjoa169f1+jDT4k+7sU9q7u02bx5s1AqlSIsLExcunRJDB06VFhYWIiEhAQhhBB9+/YVX375pVT/6NGjQk9PT/zwww/i8uXLIigoSOjr64vz589LdWbPni0sLCzErl27xF9//SU6duwonJ2dxYsXL4q8f2+jbd9nz54tDAwMxLZt28T9+/el29OnT4UQQjx9+lSMHz9exMTEiPj4eHHgwAFRv359UaVKFfHy5cti6WNetO17cHCwiIiIEDdu3BCxsbGiZ8+ewtDQUFy8eFGq86Eed5WmTZuKHj165CgvLcf96dOn4uzZs+Ls2bMCgJg/f744e/asuHXrlhBCiC+//FL07dtXqv/PP/8IY2NjMWHCBHH58mWxdOlSoaurK/bt2yfVede+LCm07fuGDRuEnp6eWLp0qdprPTk5Waozbtw4ERUVJeLj48XRo0eFt7e3sLKyEklJSUXev7fRtu8hISFi586d4tq1a+L8+fNi9OjRQkdHRxw4cECq86Eed5XPP/9ceHh45LrO0nLchRCCgSgfFi9eLCpVqiQMDAxEo0aNxPHjx6VlXl5ewt/fX63+1q1bRdWqVYWBgYGoUaOGCA8PV1uenZ0tvv32W2FrayuUSqVo1aqVuHr1alF0RWva9N3R0VEAyHELCgoSQgjx/Plz0aZNG2FtbS309fWFo6OjGDJkSIl7k1DRpu9jxoyR6tra2opPP/1UnDlzRm19H+pxF0KIK1euCAAiMjIyx7pKy3FXnU795k3VV39/f+Hl5ZXjMXXr1hUGBgaicuXKYs2aNTnW+7Z9WVJo23cvL6+31hfi1SUIypcvLwwMDESFChVEjx49xPXr14u2YxrQtu/ff/+9cHFxEYaGhsLS0lI0b95cHDp0KMd6P8TjLsSry4UYGRmJFStW5LrO0nLchRBCIYQQhTwIRURERFSicQ4RERERyR4DEREREckeAxERERHJHgMRERERyR4DEREREckeAxERERHJHgMRERERyR4DERF90Jo3b44xY8YUdzNyFRYWBgsLi3fWUygU2LlzZ6G3h6g4HDlyBO3bt4e9vX2+n+sRERFo3LgxypQpA2tra3Tp0gU3b97Uah0MRESUp4SEBIwePRqurq4wNDSEra0tmjRpguXLl+P58+fF3TyNbN++HdOnT8/345s3bw6FQgGFQgFDQ0O4u7tj2bJlBdK2Hj164O+//5buT506FXXr1s1R7/79+2q/Gk70IXn27Bnq1KmDpUuX5uvx8fHx6NixI1q2bIm4uDhERETgv//+Q+fOnbVaj16+tk5EH7x//vkHTZo0gYWFBb777jvUqlULSqUS58+fx4oVK1ChQgV06NChuJv5TpaWlu+9jiFDhmDatGl4/vw5fv75ZwQEBKBs2bLo1avXe63XyMgIRkZG76z3ofwqOlFu2rVr99bAn5aWhq+//hqbNm1CcnIyatasie+//x7NmzcHAMTGxiIrKwszZsyAjs6rcZ7x48ejY8eOyMjIgL6+vkbt4AgREeVq+PDh0NPTw+nTp9G9e3e4ubmhcuXK6NixI8LDw9G+fXupbnJyMgYPHgxra2uYmZmhZcuWOHfunLRcNfKxbt06ODk5wdzcHD179sTTp0+lOmlpaRg1ahRsbGxgaGiIpk2b4tSpU9LyqKgoKBQKREREoF69ejAyMkLLli2RlJSEvXv3ws3NDWZmZujdu7fa6NWbX5mlpaVh0qRJcHBwgFKphKurK1atWvXWfWFsbAw7OztUrlwZU6dORZUqVfDbb78BAG7fvo2OHTvC1NQUZmZm6N69OxITE6XHnjt3Di1atECZMmVgZmaGBg0a4PTp0wDUvzILCwtDcHAwzp07J41IhYWFAcj5ldn58+fRsmVLGBkZoVy5chg6dChSU1Ol5f3794efnx9++OEHlC9fHuXKlUNAQAAyMjLe2k+ikmjEiBGIiYnB5s2b8ddff6Fbt25o27Ytrl27BgBo0KABdHR0sGbNGmRlZeHJkydYt24dvL29NQ5DAAMREeXi4cOHiIyMREBAAExMTHKto1AopP9369ZNCiaxsbGoX78+WrVqhUePHkl1bty4gZ07d2L37t3YvXs3oqOjMXv2bGn5xIkT8euvv2Lt2rU4c+YMXF1d4ePjo7YO4FW4WrJkCY4dO4Y7d+6ge/fuWLBgATZu3Ijw8HBERkZi8eLFefatX79+2LRpExYtWoTLly/jxx9/hKmpqVb7x8jICOnp6cjOzkbHjh3x6NEjREdHY//+/fjnn3/Qo0cPqW6fPn1QsWJFnDp1CrGxsfjyyy9zfZPu0aMHxo0bhxo1auD+/fu4f/++2npUnj17Bh8fH5QtWxanTp3CL7/8ggMHDmDEiBFq9Q4fPowbN27g8OHDWLt2LcLCwqSARVRa3L59G2vWrMEvv/yCTz75BC4uLhg/fjyaNm2KNWvWAACcnZ0RGRmJr776CkqlEhYWFrh79y62bt2q3caK+9dliajkOX78uAAgtm/frlZerlw5YWJiIkxMTMTEiROFEEL88ccfwszMTLx8+VKtrouLi/jxxx+FEEIEBQUJY2NjkZKSIi2fMGGC8PDwEEIIkZqaKvT19cWGDRuk5enp6cLe3l7MmTNHCPF/v8R94MABqc6sWbMEAHHjxg2p7H//+5/w8fGR7nt5eYnRo0cLIYS4evWqACD279+v8b54/fGZmZli3bp1AoBYsmSJiIyMFLq6uuL27dtS/YsXLwoA4uTJk0IIIcqUKSPCwsJyXfeaNWuEubm5dD8oKEjUqVMnRz0AYseOHUIIIVasWCHKli0rUlNTpeXh4eFCR0dHJCQkCCFe/Sq5o6OjyMzMlOp069ZN9OjRQ+N+ExWH15/rQgixe/duAUB631Hd9PT0RPfu3YUQQty/f19UqVJFTJgwQZw5c0ZER0cLLy8v0apVK5Gdna3xtjmHiIg0dvLkSWRnZ6NPnz5IS0sD8OorodTUVJQrV06t7osXL3Djxg3pvpOTE8qUKSPdL1++PJKSkgC8Gj3KyMhAkyZNpOX6+vpo1KgRLl++rLbe2rVrS/+3tbWFsbExKleurFZ28uTJXNsfFxcHXV1deHl5adXvZcuWYeXKlUhPT4euri7Gjh2LYcOGYcmSJXBwcICDg4NU193dHRYWFrh8+TI++ugjBAYGYvDgwdIQfrdu3eDi4qLV9l93+fJl1KlTR23krkmTJsjOzsbVq1dha2sLAKhRowZ0dXWlOuXLl8f58+fzvV2i4pCamgpdXV3ExsaqPZ8BSCO7S5cuhbm5OebMmSMtW79+PRwcHHDixAk0btxYo20xEBFRDq6urlAoFLh69apauSp4vD4RODU1FeXLl0dUVFSO9bx+SvmbXxMpFApkZ2dr3bbX16NQKLRaryYTmHPTp08ffP311zAyMkL58uWliZuamDp1Knr37o3w8HDs3bsXQUFB2Lx5Mzp16pSvtmiqoPY3UXGqV68esrKykJSUhE8++STXOs+fP8/xmlSFJ22e85xDREQ5lCtXDq1bt8aSJUvw7Nmzt9atX78+EhISoKenB1dXV7WblZWVRttzcXGBgYEBjh49KpVlZGTg1KlTcHd3f6++vK5WrVrIzs5GdHS0Vo8zNzeHq6srKlSooPbG6+bmhjt37uDOnTtS2aVLl5CcnKzW7qpVq2Ls2LGIjIxE586dpbkPbzIwMEBWVtZb2+Lm5oZz586pHZejR49CR0cH1apV06pfRCVBamoq4uLiEBcXB+DVafRxcXG4ffs2qlatij59+qBfv37Yvn074uPjcfLkScyaNQvh4eEAAF9fX5w6dQrTpk3DtWvXcObMGQwYMACOjo6oV6+exu1gICKiXC1btgyZmZlo2LAhtmzZgsuXL+Pq1atYv349rly5In0C8/b2hqenJ/z8/BAZGYmbN2/i2LFj+Prrr6Wzqd7FxMQEw4YNw4QJE7Bv3z5cunQJQ4YMwfPnzzFo0KAC65OTkxP8/f0xcOBA7Ny5E/Hx8YiKitJ+8uX/5+3tjVq1aqFPnz44c+YMTp48iX79+sHLywsNGzbEixcvMGLECERFReHWrVs4evQoTp06BTc3tzzbp/pj8N9//0lfS76uT58+MDQ0hL+/Py5cuIDDhw9j5MiR6Nu3r/R1GVFpcvr0adSrV08KL4GBgahXrx6mTJkCAFizZg369euHcePGoVq1avDz88OpU6dQqVIlAEDLli2xceNG7Ny5E/Xq1UPbtm2hVCqxb98+rUaF+ZUZEeXKxcUFZ8+exXfffYfJkyfj7t27UCqVcHd3x/jx4zF8+HAAr76K2bNnD77++msMGDAADx48gJ2dHZo1a6bVH+jZs2cjOzsbffv2xdOnT9GwYUNERESgbNmyBdqv5cuX46uvvsLw4cPx8OFDVKpUCV999VW+1qVQKLBr1y6MHDkSzZo1g46ODtq2bSud5aarq4uHDx+iX79+SExMhJWVFTp37ozg4OBc19elSxds374dLVq0QHJyMtasWYP+/fur1TE2NkZERARGjx6Njz76CMbGxujSpQvmz5+frz4QFbfmzZvj1Xzq3Onr6yM4ODjP1w0A9OzZEz179nyvdijE21pBREREJAP8yoyIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGSPgYiIiIhkj4GIiIiIZI+BiIiIiGTv/wFhPLivKc6GgAAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "plt.hist(off_target_mutations_df['POS'], bins=50)\n",
    "plt.xlabel('Genomic Position')\n",
    "plt.ylabel('Mutation Count')\n",
    "plt.title('Distribution of Off-Target Mutations on Chromosome 2')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 424
    },
    "id": "G77aP_CHlOb_",
    "outputId": "a71e7bcb-4012-4321-e2a9-aeb514a296d6"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"off_target_mutations_df\",\n  \"rows\": 70674,\n  \"fields\": [\n    {\n      \"column\": \"CHROM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chr2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"POS\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 54989493,\n        \"min\": 3051590,\n        \"max\": 182002086,\n        \"num_unique_values\": 70665,\n        \"samples\": [\n          150306517\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"REF\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 482,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ALT\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 545,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"QUAL\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 55.71049885114257,\n        \"min\": 3.0108699798583984,\n        \"max\": 228.41400146484372,\n        \"num_unique_values\": 29707,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"FILTER\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GENOTYPE\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 1,\n        \"max\": 3,\n        \"num_unique_values\": 2,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "off_target_mutations_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-3ca3ba04-5ff3-4920-9622-38872a650d12\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENOTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>52.203800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>45.253502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>47.249500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>55.332600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3051635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>59.256699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70678</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70679</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70680</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>40.932098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70681</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>30.957300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70682</th>\n",
       "      <td>chr2</td>\n",
       "      <td>182002086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>5.826790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70674 rows × 8 columns</p>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-3ca3ba04-5ff3-4920-9622-38872a650d12')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-3ca3ba04-5ff3-4920-9622-38872a650d12 button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-3ca3ba04-5ff3-4920-9622-38872a650d12');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-15e70d7d-29d5-40a4-a950-8c7b3cc12530\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-15e70d7d-29d5-40a4-a950-8c7b3cc12530')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-15e70d7d-29d5-40a4-a950-8c7b3cc12530 button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_420a7258-2416-4948-9854-81fa7b960eab\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('off_target_mutations_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_420a7258-2416-4948-9854-81fa7b960eab button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('off_target_mutations_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "      CHROM        POS  ID REF ALT       QUAL  FILTER  GENOTYPE\n",
       "0      chr2    3051590 NaN   T   C  52.203800     NaN         1\n",
       "1      chr2    3051604 NaN   A   C  45.253502     NaN         1\n",
       "2      chr2    3051621 NaN   T   C  47.249500     NaN         1\n",
       "3      chr2    3051632 NaN   A   G  55.332600     NaN         1\n",
       "4      chr2    3051635 NaN   A   G  59.256699     NaN         1\n",
       "...     ...        ...  ..  ..  ..        ...     ...       ...\n",
       "70678  chr2  182002062 NaN   A   T  30.957300     NaN         1\n",
       "70679  chr2  182002065 NaN   G   A  40.932098     NaN         3\n",
       "70680  chr2  182002067 NaN   G   A  40.932098     NaN         3\n",
       "70681  chr2  182002068 NaN   A   T  30.957300     NaN         1\n",
       "70682  chr2  182002086 NaN   T   A   5.826790     NaN         1\n",
       "\n",
       "[70674 rows x 8 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "off_target_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 143
    },
    "id": "BRSY0G23lPHW",
    "outputId": "44e34520-4875-4def-f518-3a9f0dfe03f7"
   },
   "outputs": [
    {
     "data": {
      "application/vnd.google.colaboratory.intrinsic+json": {
       "summary": "{\n  \"name\": \"off_target_summary_df\",\n  \"rows\": 3,\n  \"fields\": [\n    {\n      \"column\": \"mutation_count\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 70674.0,\n        \"max\": 70674.0,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          70674.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"genotype_counts\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 13476.041035853223,\n        \"min\": 25808.0,\n        \"max\": 44866.0,\n        \"num_unique_values\": 2,\n        \"samples\": [\n          25808.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
       "type": "dataframe",
       "variable_name": "off_target_summary_df"
      },
      "text/html": [
       "\n",
       "  <div id=\"df-92c245ae-fcca-4763-9a78-e71ecff88b48\" class=\"colab-df-container\">\n",
       "    <div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mutation_count</th>\n",
       "      <th>genotype_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr2</th>\n",
       "      <td>70674.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>(chr2, 1)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>44866.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>(chr2, 3)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>25808.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "    <div class=\"colab-df-buttons\">\n",
       "\n",
       "  <div class=\"colab-df-container\">\n",
       "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-92c245ae-fcca-4763-9a78-e71ecff88b48')\"\n",
       "            title=\"Convert this dataframe to an interactive table.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
       "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "\n",
       "  <style>\n",
       "    .colab-df-container {\n",
       "      display:flex;\n",
       "      gap: 12px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert {\n",
       "      background-color: #E8F0FE;\n",
       "      border: none;\n",
       "      border-radius: 50%;\n",
       "      cursor: pointer;\n",
       "      display: none;\n",
       "      fill: #1967D2;\n",
       "      height: 32px;\n",
       "      padding: 0 0 0 0;\n",
       "      width: 32px;\n",
       "    }\n",
       "\n",
       "    .colab-df-convert:hover {\n",
       "      background-color: #E2EBFA;\n",
       "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "      fill: #174EA6;\n",
       "    }\n",
       "\n",
       "    .colab-df-buttons div {\n",
       "      margin-bottom: 4px;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert {\n",
       "      background-color: #3B4455;\n",
       "      fill: #D2E3FC;\n",
       "    }\n",
       "\n",
       "    [theme=dark] .colab-df-convert:hover {\n",
       "      background-color: #434B5C;\n",
       "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "      fill: #FFFFFF;\n",
       "    }\n",
       "  </style>\n",
       "\n",
       "    <script>\n",
       "      const buttonEl =\n",
       "        document.querySelector('#df-92c245ae-fcca-4763-9a78-e71ecff88b48 button.colab-df-convert');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      async function convertToInteractive(key) {\n",
       "        const element = document.querySelector('#df-92c245ae-fcca-4763-9a78-e71ecff88b48');\n",
       "        const dataTable =\n",
       "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
       "                                                    [key], {});\n",
       "        if (!dataTable) return;\n",
       "\n",
       "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
       "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
       "          + ' to learn more about interactive tables.';\n",
       "        element.innerHTML = '';\n",
       "        dataTable['output_type'] = 'display_data';\n",
       "        await google.colab.output.renderOutput(dataTable, element);\n",
       "        const docLink = document.createElement('div');\n",
       "        docLink.innerHTML = docLinkHtml;\n",
       "        element.appendChild(docLink);\n",
       "      }\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div id=\"df-c8355285-d2a0-4c21-a26d-ee7176dce76d\">\n",
       "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c8355285-d2a0-4c21-a26d-ee7176dce76d')\"\n",
       "            title=\"Suggest charts\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "     width=\"24px\">\n",
       "    <g>\n",
       "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
       "    </g>\n",
       "</svg>\n",
       "  </button>\n",
       "\n",
       "<style>\n",
       "  .colab-df-quickchart {\n",
       "      --bg-color: #E8F0FE;\n",
       "      --fill-color: #1967D2;\n",
       "      --hover-bg-color: #E2EBFA;\n",
       "      --hover-fill-color: #174EA6;\n",
       "      --disabled-fill-color: #AAA;\n",
       "      --disabled-bg-color: #DDD;\n",
       "  }\n",
       "\n",
       "  [theme=dark] .colab-df-quickchart {\n",
       "      --bg-color: #3B4455;\n",
       "      --fill-color: #D2E3FC;\n",
       "      --hover-bg-color: #434B5C;\n",
       "      --hover-fill-color: #FFFFFF;\n",
       "      --disabled-bg-color: #3B4455;\n",
       "      --disabled-fill-color: #666;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart {\n",
       "    background-color: var(--bg-color);\n",
       "    border: none;\n",
       "    border-radius: 50%;\n",
       "    cursor: pointer;\n",
       "    display: none;\n",
       "    fill: var(--fill-color);\n",
       "    height: 32px;\n",
       "    padding: 0;\n",
       "    width: 32px;\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart:hover {\n",
       "    background-color: var(--hover-bg-color);\n",
       "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "    fill: var(--button-hover-fill-color);\n",
       "  }\n",
       "\n",
       "  .colab-df-quickchart-complete:disabled,\n",
       "  .colab-df-quickchart-complete:disabled:hover {\n",
       "    background-color: var(--disabled-bg-color);\n",
       "    fill: var(--disabled-fill-color);\n",
       "    box-shadow: none;\n",
       "  }\n",
       "\n",
       "  .colab-df-spinner {\n",
       "    border: 2px solid var(--fill-color);\n",
       "    border-color: transparent;\n",
       "    border-bottom-color: var(--fill-color);\n",
       "    animation:\n",
       "      spin 1s steps(1) infinite;\n",
       "  }\n",
       "\n",
       "  @keyframes spin {\n",
       "    0% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "      border-left-color: var(--fill-color);\n",
       "    }\n",
       "    20% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    30% {\n",
       "      border-color: transparent;\n",
       "      border-left-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    40% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-top-color: var(--fill-color);\n",
       "    }\n",
       "    60% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "    }\n",
       "    80% {\n",
       "      border-color: transparent;\n",
       "      border-right-color: var(--fill-color);\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "    90% {\n",
       "      border-color: transparent;\n",
       "      border-bottom-color: var(--fill-color);\n",
       "    }\n",
       "  }\n",
       "</style>\n",
       "\n",
       "  <script>\n",
       "    async function quickchart(key) {\n",
       "      const quickchartButtonEl =\n",
       "        document.querySelector('#' + key + ' button');\n",
       "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
       "      try {\n",
       "        const charts = await google.colab.kernel.invokeFunction(\n",
       "            'suggestCharts', [key], {});\n",
       "      } catch (error) {\n",
       "        console.error('Error during call to suggestCharts:', error);\n",
       "      }\n",
       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
       "    }\n",
       "    (() => {\n",
       "      let quickchartButtonEl =\n",
       "        document.querySelector('#df-c8355285-d2a0-4c21-a26d-ee7176dce76d button');\n",
       "      quickchartButtonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "    })();\n",
       "  </script>\n",
       "</div>\n",
       "\n",
       "  <div id=\"id_ef650112-8134-4cea-9219-8c2c83c2353b\">\n",
       "    <style>\n",
       "      .colab-df-generate {\n",
       "        background-color: #E8F0FE;\n",
       "        border: none;\n",
       "        border-radius: 50%;\n",
       "        cursor: pointer;\n",
       "        display: none;\n",
       "        fill: #1967D2;\n",
       "        height: 32px;\n",
       "        padding: 0 0 0 0;\n",
       "        width: 32px;\n",
       "      }\n",
       "\n",
       "      .colab-df-generate:hover {\n",
       "        background-color: #E2EBFA;\n",
       "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
       "        fill: #174EA6;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate {\n",
       "        background-color: #3B4455;\n",
       "        fill: #D2E3FC;\n",
       "      }\n",
       "\n",
       "      [theme=dark] .colab-df-generate:hover {\n",
       "        background-color: #434B5C;\n",
       "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
       "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
       "        fill: #FFFFFF;\n",
       "      }\n",
       "    </style>\n",
       "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('off_target_summary_df')\"\n",
       "            title=\"Generate code using this dataframe.\"\n",
       "            style=\"display:none;\">\n",
       "\n",
       "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
       "       width=\"24px\">\n",
       "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
       "  </svg>\n",
       "    </button>\n",
       "    <script>\n",
       "      (() => {\n",
       "      const buttonEl =\n",
       "        document.querySelector('#id_ef650112-8134-4cea-9219-8c2c83c2353b button.colab-df-generate');\n",
       "      buttonEl.style.display =\n",
       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
       "\n",
       "      buttonEl.onclick = () => {\n",
       "        google.colab.notebook.generateWithVariable('off_target_summary_df');\n",
       "      }\n",
       "      })();\n",
       "    </script>\n",
       "  </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n"
      ],
      "text/plain": [
       "           mutation_count  genotype_counts\n",
       "chr2              70674.0              NaN\n",
       "(chr2, 1)             NaN          44866.0\n",
       "(chr2, 3)             NaN          25808.0"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "off_target_summary_df = off_target_mutations_df.groupby('CHROM').agg(\n",
    "    mutation_count=('POS', 'count'),\n",
    "    genotype_counts=('GENOTYPE', 'value_counts')\n",
    ")\n",
    "off_target_summary_df"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "name": "python3"
  },
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
